<a id="item31"></a>


## Regression with Keras assignment: In this part we use the normalized data instead of the original raw data for training


In [4]:
# All Libraries required for this lab are listed below. The libraries pre-installed on Skills Network Labs are commented. 
# If you run this notebook on a different environment, e.g. your desktop, you may need to uncomment and install certain libraries.

#!pip install numpy==2.0.2
#!pip install pandas==2.2.2
#!pip install tensorflow_cpu==2.18.0


     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.9/60.9 kB 6.7 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 19.5/19.5 MB 88.7 MB/s eta 0:00:00:00:0100:01
  Attempting uninstall: numpy
    Found existing installation: numpy 2.2.1
    Uninstalling numpy-2.2.1:
      Successfully uninstalled numpy-2.2.1
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 47.8/47.8 kB 6.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 230.2/230.2 MB 2.1 MB/s eta 0:00:00:00:0100:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.7/133.7 kB 79.7 kB/s eta 0:00:0000:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.5/57.5 kB 7.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.9/5.9 MB 64.9 MB/s eta 0:00:00:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.5/5.5 MB 52.5 MB/s eta 0:00:00:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.3/1.3 MB 67.7 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 24.5/24.5 MB 23.9 MB/s eta 0:00:00:00:0100:01
   ━━━━━━━━━━━

## Importing the required libraries

In [1]:
import pandas as pd
import numpy as np
import keras

import warnings
warnings.simplefilter('ignore', FutureWarning)

In [2]:
from keras.models import Sequential
from keras.layers import Dense
from keras.layers import Input

In [3]:
from sklearn.model_selection import train_test_split

In [4]:
from sklearn.metrics import mean_squared_error

In [24]:
from sklearn.preprocessing import StandardScaler, MinMaxScaler, RobustScaler

## Read the data

In [5]:
filepath='concrete_data.csv'
concrete_data = pd.read_csv(filepath)

concrete_data.head()

Cement  Blast Furnace Slag  Fly Ash  Water  Superplasticizer  \
0   540.0                 0.0      0.0  162.0               2.5   
1   540.0                 0.0      0.0  162.0               2.5   
2   332.5               142.5      0.0  228.0               0.0   
3   332.5               142.5      0.0  228.0               0.0   
4   198.6               132.4      0.0  192.0               0.0   

   Coarse Aggregate  Fine Aggregate  Age  Strength  
0            1040.0           676.0   28     79.99  
1            1055.0           676.0   28     61.89  
2             932.0           594.0  270     40.27  
3             932.0           594.0  365     41.05  
4             978.4           825.5  360     44.30

In [6]:
concrete_data.describe()

Cement  Blast Furnace Slag      Fly Ash        Water  \
count  1030.000000         1030.000000  1030.000000  1030.000000   
mean    281.167864           73.895825    54.188350   181.567282   
std     104.506364           86.279342    63.997004    21.354219   
min     102.000000            0.000000     0.000000   121.800000   
25%     192.375000            0.000000     0.000000   164.900000   
50%     272.900000           22.000000     0.000000   185.000000   
75%     350.000000          142.950000   118.300000   192.000000   
max     540.000000          359.400000   200.100000   247.000000   

       Superplasticizer  Coarse Aggregate  Fine Aggregate          Age  \
count       1030.000000       1030.000000     1030.000000  1030.000000   
mean           6.204660        972.918932      773.580485    45.662136   
std            5.973841         77.753954       80.175980    63.169912   
min            0.000000        801.000000      594.000000     1.000000   
25%            0.000000        932.000000      730.950000     7.000000   
50%            6.400000        968.000000      779.500000    28.000000   
75%           10.200000       1029.400000      824.000000    56.000000   
max           32.200000       1145.000000      992.600000   365.000000   

          Strength  
count  1030.000000  
mean     35.817961  
std      16.705742  
min       2.330000  
25%      23.710000  
50%      34.445000  
75%      46.135000  
max      82.600000

In [7]:
concrete_data.isnull().sum()

Cement                0
Blast Furnace Slag    0
Fly Ash               0
Water                 0
Superplasticizer      0
Coarse Aggregate      0
Fine Aggregate        0
Age                   0
Strength              0
dtype: int64

## Separating the predictors and the target

In [8]:
concrete_data_columns = concrete_data.columns

In [9]:
predictors = concrete_data[concrete_data_columns[concrete_data_columns != 'Strength']] # all columns except Strength
target = concrete_data['Strength'] # Strength column

<a id="item2"></a>


## Normalizing the data

In [10]:
predictors_normalized = (predictors - predictors.mean()) / predictors.std()
predictors_normalized.head()

Cement  Blast Furnace Slag   Fly Ash     Water  Superplasticizer  \
0  2.476712           -0.856472 -0.846733 -0.916319         -0.620147   
1  2.476712           -0.856472 -0.846733 -0.916319         -0.620147   
2  0.491187            0.795140 -0.846733  2.174405         -1.038638   
3  0.491187            0.795140 -0.846733  2.174405         -1.038638   
4 -0.790075            0.678079 -0.846733  0.488555         -1.038638   

   Coarse Aggregate  Fine Aggregate       Age  
0          0.862735       -1.217079 -0.279597  
1          1.055651       -1.217079 -0.279597  
2         -0.526262       -2.239829  3.551340  
3         -0.526262       -2.239829  5.055221  
4          0.070492        0.647569  4.976069

## Train-test splitting of the data

In [13]:
X_train, X_test, y_train, y_test = train_test_split(predictors_normalized, target, test_size=0.30, random_state=42)

## Building a neural network with keras using 1 hidden layer of 10 nodes,  ReLu activation function, adam optimizer and the mean squared error  as the loss function.

In [14]:
def regression_model():
    # create model
    model = Sequential()
    model.add(Input(shape=(predictors.shape[1],))) # input layer with number of predictors
    model.add(Dense(10, activation='relu'))
    
    model.add(Dense(1))
    
    # compile model
    model.compile(optimizer='adam', loss='mean_squared_error')
    return model

## Model initialization

In [15]:
model = regression_model()

## Model training by fitting the model instance on the training data

In [16]:
model.fit(X_train, y_train, validation_split=0.3, epochs=50, verbose=2)

Epoch 1/50
16/16 - 9s - loss: 1576.8514 - val_loss: 1453.2886 - 9s/epoch - 538ms/step
Epoch 2/50
16/16 - 0s - loss: 1565.9700 - val_loss: 1443.2518 - 204ms/epoch - 13ms/step
Epoch 3/50
16/16 - 0s - loss: 1555.0278 - val_loss: 1432.7581 - 206ms/epoch - 13ms/step
Epoch 4/50
16/16 - 0s - loss: 1543.4423 - val_loss: 1421.9697 - 167ms/epoch - 10ms/step
Epoch 5/50
16/16 - 0s - loss: 1531.4397 - val_loss: 1410.7814 - 152ms/epoch - 10ms/step
Epoch 6/50
16/16 - 0s - loss: 1518.8342 - val_loss: 1399.1389 - 151ms/epoch - 9ms/step
Epoch 7/50
16/16 - 0s - loss: 1505.9224 - val_loss: 1386.6875 - 173ms/epoch - 11ms/step
Epoch 8/50
16/16 - 0s - loss: 1491.7512 - val_loss: 1374.1271 - 185ms/epoch - 12ms/step
Epoch 9/50
16/16 - 0s - loss: 1477.5569 - val_loss: 1360.5216 - 170ms/epoch - 11ms/step
Epoch 10/50
16/16 - 0s - loss: 1462.1027 - val_loss: 1346.2874 - 196ms/epoch - 12ms/step
Epoch 11/50
16/16 - 0s - loss: 1445.7618 - val_loss: 1331.4727 - 235ms/epoch - 15ms/step
Epoch 12/50
16/16 - 0s - loss: 14

## Calculating model predictions over the test data using model.predict()

In [17]:
predicted_strength = model.predict(X_test)

10/10 [==============================] - 0s 5ms/step


In [18]:
#predicted_strength

## Calculating mean squared error from mean_squared_error of scikit learn

In [19]:
mse = mean_squared_error(y_test, predicted_strength)

In [20]:
print('Mean squared error on the test data is: ', mse)

Mean squared error on the test data is:  464.2052093003634


## Repeating, splitting and model trainining 50 times and calculating the mse and standard deviation of MSE

In [21]:
mse_50 = []
model = regression_model()
for i in range(50):
    X_train, X_test, y_train, y_test = train_test_split(predictors, target, test_size=0.30)
    model.fit(X_train, y_train, validation_split=0.3, epochs=50, verbose=2)
    predictions = model.predict(X_test)
    mse_50.append(mean_squared_error(y_test, predictions))

Epoch 1/50
16/16 - 1s - loss: 100162.2188 - val_loss: 68771.9688 - 1s/epoch - 68ms/step
Epoch 2/50
16/16 - 0s - loss: 50872.4258 - val_loss: 31969.9121 - 212ms/epoch - 13ms/step
Epoch 3/50
16/16 - 0s - loss: 22651.2539 - val_loss: 13028.9697 - 206ms/epoch - 13ms/step
Epoch 4/50
16/16 - 0s - loss: 9063.3203 - val_loss: 5023.5503 - 194ms/epoch - 12ms/step
Epoch 5/50
16/16 - 0s - loss: 3657.3345 - val_loss: 2400.4485 - 196ms/epoch - 12ms/step
Epoch 6/50
16/16 - 0s - loss: 2064.7095 - val_loss: 1760.0339 - 261ms/epoch - 16ms/step
Epoch 7/50
16/16 - 0s - loss: 1682.4419 - val_loss: 1669.2549 - 240ms/epoch - 15ms/step
Epoch 8/50
16/16 - 0s - loss: 1623.2440 - val_loss: 1650.9170 - 253ms/epoch - 16ms/step
Epoch 9/50
16/16 - 0s - loss: 1590.1654 - val_loss: 1620.3264 - 238ms/epoch - 15ms/step
Epoch 10/50
16/16 - 0s - loss: 1560.1233 - val_loss: 1585.6157 - 217ms/epoch - 14ms/step
Epoch 11/50
16/16 - 0s - loss: 1529.1608 - val_loss: 1553.0623 - 209ms/epoch - 13ms/step
Epoch 12/50
16/16 - 0s - l

16/16 - 0s - loss: 261.9138 - val_loss: 261.3057 - 193ms/epoch - 12ms/step
Epoch 45/50
16/16 - 0s - loss: 257.2480 - val_loss: 255.5474 - 211ms/epoch - 13ms/step
Epoch 46/50
16/16 - 0s - loss: 253.8419 - val_loss: 252.4402 - 218ms/epoch - 14ms/step
Epoch 47/50
16/16 - 0s - loss: 249.0724 - val_loss: 245.8758 - 216ms/epoch - 14ms/step
Epoch 48/50
16/16 - 0s - loss: 243.9901 - val_loss: 244.4630 - 218ms/epoch - 14ms/step
Epoch 49/50
16/16 - 0s - loss: 240.8369 - val_loss: 238.0315 - 228ms/epoch - 14ms/step
Epoch 50/50
16/16 - 0s - loss: 236.5787 - val_loss: 234.4917 - 237ms/epoch - 15ms/step
10/10 [==============================] - 0s 4ms/step
Epoch 1/50
16/16 - 0s - loss: 226.8745 - val_loss: 243.3857 - 329ms/epoch - 21ms/step
Epoch 2/50
16/16 - 0s - loss: 222.8928 - val_loss: 240.1196 - 225ms/epoch - 14ms/step
Epoch 3/50
16/16 - 0s - loss: 219.1804 - val_loss: 235.9998 - 220ms/epoch - 14ms/step
Epoch 4/50
16/16 - 0s - loss: 216.4582 - val_loss: 232.7496 - 227ms/epoch - 14ms/step
Epoch 

Epoch 38/50
16/16 - 0s - loss: 115.9241 - val_loss: 106.9307 - 199ms/epoch - 12ms/step
Epoch 39/50
16/16 - 0s - loss: 116.3696 - val_loss: 106.0029 - 202ms/epoch - 13ms/step
Epoch 40/50
16/16 - 0s - loss: 115.0432 - val_loss: 105.7276 - 210ms/epoch - 13ms/step
Epoch 41/50
16/16 - 0s - loss: 114.9579 - val_loss: 105.1438 - 202ms/epoch - 13ms/step
Epoch 42/50
16/16 - 0s - loss: 115.0847 - val_loss: 106.1655 - 219ms/epoch - 14ms/step
Epoch 43/50
16/16 - 0s - loss: 114.8394 - val_loss: 104.5487 - 210ms/epoch - 13ms/step
Epoch 44/50
16/16 - 0s - loss: 114.0745 - val_loss: 104.9873 - 200ms/epoch - 12ms/step
Epoch 45/50
16/16 - 0s - loss: 114.4854 - val_loss: 105.2788 - 200ms/epoch - 12ms/step
Epoch 46/50
16/16 - 0s - loss: 113.9775 - val_loss: 104.2510 - 204ms/epoch - 13ms/step
Epoch 47/50
16/16 - 0s - loss: 113.6248 - val_loss: 103.7637 - 248ms/epoch - 16ms/step
Epoch 48/50
16/16 - 0s - loss: 113.6486 - val_loss: 103.2085 - 205ms/epoch - 13ms/step
Epoch 49/50
16/16 - 0s - loss: 113.2320 - v

Epoch 32/50
16/16 - 0s - loss: 107.9067 - val_loss: 115.1437 - 241ms/epoch - 15ms/step
Epoch 33/50
16/16 - 0s - loss: 110.7762 - val_loss: 118.8542 - 228ms/epoch - 14ms/step
Epoch 34/50
16/16 - 0s - loss: 107.6005 - val_loss: 113.0602 - 191ms/epoch - 12ms/step
Epoch 35/50
16/16 - 0s - loss: 105.5715 - val_loss: 114.2976 - 197ms/epoch - 12ms/step
Epoch 36/50
16/16 - 0s - loss: 106.1063 - val_loss: 117.9204 - 204ms/epoch - 13ms/step
Epoch 37/50
16/16 - 0s - loss: 105.9266 - val_loss: 109.9654 - 244ms/epoch - 15ms/step
Epoch 38/50
16/16 - 0s - loss: 104.8553 - val_loss: 109.9550 - 246ms/epoch - 15ms/step
Epoch 39/50
16/16 - 0s - loss: 104.9417 - val_loss: 116.2169 - 220ms/epoch - 14ms/step
Epoch 40/50
16/16 - 0s - loss: 107.4512 - val_loss: 114.6328 - 164ms/epoch - 10ms/step
Epoch 41/50
16/16 - 0s - loss: 107.7811 - val_loss: 111.7150 - 175ms/epoch - 11ms/step
Epoch 42/50
16/16 - 0s - loss: 106.5751 - val_loss: 109.8214 - 184ms/epoch - 12ms/step
Epoch 43/50
16/16 - 0s - loss: 105.2215 - v

Epoch 26/50
16/16 - 0s - loss: 101.7464 - val_loss: 150.2930 - 190ms/epoch - 12ms/step
Epoch 27/50
16/16 - 0s - loss: 101.7024 - val_loss: 144.2368 - 199ms/epoch - 12ms/step
Epoch 28/50
16/16 - 0s - loss: 107.8131 - val_loss: 152.5747 - 198ms/epoch - 12ms/step
Epoch 29/50
16/16 - 0s - loss: 103.5343 - val_loss: 140.2283 - 205ms/epoch - 13ms/step
Epoch 30/50
16/16 - 0s - loss: 101.5322 - val_loss: 138.9323 - 249ms/epoch - 16ms/step
Epoch 31/50
16/16 - 0s - loss: 102.1079 - val_loss: 138.5531 - 216ms/epoch - 13ms/step
Epoch 32/50
16/16 - 0s - loss: 104.7251 - val_loss: 138.6004 - 208ms/epoch - 13ms/step
Epoch 33/50
16/16 - 0s - loss: 106.0066 - val_loss: 139.0594 - 208ms/epoch - 13ms/step
Epoch 34/50
16/16 - 0s - loss: 101.7382 - val_loss: 139.0306 - 203ms/epoch - 13ms/step
Epoch 35/50
16/16 - 0s - loss: 102.4203 - val_loss: 146.7082 - 220ms/epoch - 14ms/step
Epoch 36/50
16/16 - 0s - loss: 102.8291 - val_loss: 138.8024 - 200ms/epoch - 13ms/step
Epoch 37/50
16/16 - 0s - loss: 101.6768 - v

Epoch 20/50
16/16 - 0s - loss: 118.2315 - val_loss: 105.9627 - 208ms/epoch - 13ms/step
Epoch 21/50
16/16 - 0s - loss: 114.5255 - val_loss: 104.0164 - 226ms/epoch - 14ms/step
Epoch 22/50
16/16 - 0s - loss: 114.5260 - val_loss: 104.0815 - 210ms/epoch - 13ms/step
Epoch 23/50
16/16 - 0s - loss: 115.7376 - val_loss: 107.2165 - 220ms/epoch - 14ms/step
Epoch 24/50
16/16 - 0s - loss: 116.9117 - val_loss: 104.0960 - 187ms/epoch - 12ms/step
Epoch 25/50
16/16 - 0s - loss: 117.3180 - val_loss: 107.2346 - 221ms/epoch - 14ms/step
Epoch 26/50
16/16 - 0s - loss: 115.1123 - val_loss: 104.7547 - 223ms/epoch - 14ms/step
Epoch 27/50
16/16 - 0s - loss: 114.3947 - val_loss: 104.0095 - 173ms/epoch - 11ms/step
Epoch 28/50
16/16 - 0s - loss: 114.9525 - val_loss: 107.0875 - 183ms/epoch - 11ms/step
Epoch 29/50
16/16 - 0s - loss: 116.2171 - val_loss: 104.5588 - 187ms/epoch - 12ms/step
Epoch 30/50
16/16 - 0s - loss: 114.7098 - val_loss: 104.2235 - 167ms/epoch - 10ms/step
Epoch 31/50
16/16 - 0s - loss: 115.3585 - v

Epoch 14/50
16/16 - 0s - loss: 114.4605 - val_loss: 104.4667 - 192ms/epoch - 12ms/step
Epoch 15/50
16/16 - 0s - loss: 113.5529 - val_loss: 113.1370 - 220ms/epoch - 14ms/step
Epoch 16/50
16/16 - 0s - loss: 122.9544 - val_loss: 102.5342 - 209ms/epoch - 13ms/step
Epoch 17/50
16/16 - 0s - loss: 120.3907 - val_loss: 102.7231 - 200ms/epoch - 12ms/step
Epoch 18/50
16/16 - 0s - loss: 114.5289 - val_loss: 102.8512 - 199ms/epoch - 12ms/step
Epoch 19/50
16/16 - 0s - loss: 116.9634 - val_loss: 108.0360 - 209ms/epoch - 13ms/step
Epoch 20/50
16/16 - 0s - loss: 115.0891 - val_loss: 105.8718 - 202ms/epoch - 13ms/step
Epoch 21/50
16/16 - 0s - loss: 119.5822 - val_loss: 105.7888 - 200ms/epoch - 12ms/step
Epoch 22/50
16/16 - 0s - loss: 116.4606 - val_loss: 103.7600 - 201ms/epoch - 13ms/step
Epoch 23/50
16/16 - 0s - loss: 116.3886 - val_loss: 105.7786 - 210ms/epoch - 13ms/step
Epoch 24/50
16/16 - 0s - loss: 119.3994 - val_loss: 103.2047 - 206ms/epoch - 13ms/step
Epoch 25/50
16/16 - 0s - loss: 115.8045 - v

16/16 - 0s - loss: 113.0480 - val_loss: 108.7487 - 204ms/epoch - 13ms/step
Epoch 8/50
16/16 - 0s - loss: 110.3725 - val_loss: 109.5611 - 209ms/epoch - 13ms/step
Epoch 9/50
16/16 - 0s - loss: 114.1780 - val_loss: 117.5303 - 218ms/epoch - 14ms/step
Epoch 10/50
16/16 - 0s - loss: 115.5302 - val_loss: 108.5051 - 213ms/epoch - 13ms/step
Epoch 11/50
16/16 - 0s - loss: 111.1019 - val_loss: 108.5051 - 222ms/epoch - 14ms/step
Epoch 12/50
16/16 - 0s - loss: 110.1103 - val_loss: 113.2242 - 214ms/epoch - 13ms/step
Epoch 13/50
16/16 - 0s - loss: 112.1155 - val_loss: 108.7238 - 217ms/epoch - 14ms/step
Epoch 14/50
16/16 - 0s - loss: 109.3909 - val_loss: 108.6116 - 211ms/epoch - 13ms/step
Epoch 15/50
16/16 - 0s - loss: 108.7666 - val_loss: 108.3216 - 219ms/epoch - 14ms/step
Epoch 16/50
16/16 - 0s - loss: 108.5030 - val_loss: 111.2564 - 240ms/epoch - 15ms/step
Epoch 17/50
16/16 - 0s - loss: 110.1527 - val_loss: 110.6519 - 217ms/epoch - 14ms/step
Epoch 18/50
16/16 - 0s - loss: 111.5911 - val_loss: 109.1

10/10 [==============================] - 0s 4ms/step
Epoch 1/50
16/16 - 0s - loss: 107.5134 - val_loss: 116.7812 - 282ms/epoch - 18ms/step
Epoch 2/50
16/16 - 0s - loss: 105.3327 - val_loss: 117.6396 - 190ms/epoch - 12ms/step
Epoch 3/50
16/16 - 0s - loss: 104.6051 - val_loss: 125.2413 - 203ms/epoch - 13ms/step
Epoch 4/50
16/16 - 0s - loss: 105.7635 - val_loss: 118.6055 - 196ms/epoch - 12ms/step
Epoch 5/50
16/16 - 0s - loss: 112.5477 - val_loss: 117.9106 - 204ms/epoch - 13ms/step
Epoch 6/50
16/16 - 0s - loss: 109.1402 - val_loss: 119.2655 - 219ms/epoch - 14ms/step
Epoch 7/50
16/16 - 0s - loss: 105.2920 - val_loss: 128.1801 - 206ms/epoch - 13ms/step
Epoch 8/50
16/16 - 0s - loss: 106.0964 - val_loss: 130.1505 - 217ms/epoch - 14ms/step
Epoch 9/50
16/16 - 0s - loss: 104.3416 - val_loss: 141.5808 - 202ms/epoch - 13ms/step
Epoch 10/50
16/16 - 0s - loss: 107.9643 - val_loss: 123.3396 - 197ms/epoch - 12ms/step
Epoch 11/50
16/16 - 0s - loss: 104.7899 - val_loss: 121.7014 - 208ms/epoch - 13ms/step

Epoch 45/50
16/16 - 0s - loss: 120.0760 - val_loss: 129.7035 - 202ms/epoch - 13ms/step
Epoch 46/50
16/16 - 0s - loss: 124.7496 - val_loss: 116.9354 - 193ms/epoch - 12ms/step
Epoch 47/50
16/16 - 0s - loss: 119.2971 - val_loss: 136.6155 - 245ms/epoch - 15ms/step
Epoch 48/50
16/16 - 0s - loss: 112.3434 - val_loss: 116.3104 - 192ms/epoch - 12ms/step
Epoch 49/50
16/16 - 0s - loss: 109.9982 - val_loss: 116.2196 - 197ms/epoch - 12ms/step
Epoch 50/50
16/16 - 0s - loss: 113.3226 - val_loss: 116.3260 - 196ms/epoch - 12ms/step
10/10 [==============================] - 0s 4ms/step
Epoch 1/50
16/16 - 0s - loss: 113.8703 - val_loss: 111.4934 - 280ms/epoch - 18ms/step
Epoch 2/50
16/16 - 0s - loss: 114.8985 - val_loss: 122.2623 - 197ms/epoch - 12ms/step
Epoch 3/50
16/16 - 0s - loss: 113.7452 - val_loss: 107.9659 - 195ms/epoch - 12ms/step
Epoch 4/50
16/16 - 0s - loss: 109.8483 - val_loss: 109.5408 - 194ms/epoch - 12ms/step
Epoch 5/50
16/16 - 0s - loss: 110.4611 - val_loss: 112.2890 - 196ms/epoch - 12ms/

Epoch 39/50
16/16 - 0s - loss: 111.8620 - val_loss: 94.0937 - 305ms/epoch - 19ms/step
Epoch 40/50
16/16 - 0s - loss: 112.5000 - val_loss: 92.4967 - 263ms/epoch - 16ms/step
Epoch 41/50
16/16 - 0s - loss: 112.3190 - val_loss: 103.2891 - 283ms/epoch - 18ms/step
Epoch 42/50
16/16 - 0s - loss: 114.7015 - val_loss: 90.0423 - 263ms/epoch - 16ms/step
Epoch 43/50
16/16 - 0s - loss: 113.1179 - val_loss: 94.2775 - 293ms/epoch - 18ms/step
Epoch 44/50
16/16 - 0s - loss: 111.9675 - val_loss: 92.6139 - 280ms/epoch - 17ms/step
Epoch 45/50
16/16 - 0s - loss: 114.7694 - val_loss: 115.5372 - 290ms/epoch - 18ms/step
Epoch 46/50
16/16 - 0s - loss: 116.1960 - val_loss: 118.6894 - 268ms/epoch - 17ms/step
Epoch 47/50
16/16 - 0s - loss: 113.7363 - val_loss: 108.8831 - 271ms/epoch - 17ms/step
Epoch 48/50
16/16 - 0s - loss: 114.1427 - val_loss: 99.7412 - 236ms/epoch - 15ms/step
Epoch 49/50
16/16 - 0s - loss: 112.3103 - val_loss: 105.4216 - 200ms/epoch - 13ms/step
Epoch 50/50
16/16 - 0s - loss: 115.6940 - val_los

Epoch 33/50
16/16 - 0s - loss: 107.4003 - val_loss: 106.7751 - 189ms/epoch - 12ms/step
Epoch 34/50
16/16 - 0s - loss: 107.8076 - val_loss: 107.0117 - 192ms/epoch - 12ms/step
Epoch 35/50
16/16 - 0s - loss: 105.5812 - val_loss: 115.7257 - 193ms/epoch - 12ms/step
Epoch 36/50
16/16 - 0s - loss: 103.6025 - val_loss: 108.7787 - 212ms/epoch - 13ms/step
Epoch 37/50
16/16 - 0s - loss: 104.9857 - val_loss: 114.6137 - 205ms/epoch - 13ms/step
Epoch 38/50
16/16 - 0s - loss: 102.2855 - val_loss: 110.8279 - 212ms/epoch - 13ms/step
Epoch 39/50
16/16 - 0s - loss: 104.5949 - val_loss: 103.3562 - 205ms/epoch - 13ms/step
Epoch 40/50
16/16 - 0s - loss: 102.8004 - val_loss: 104.7289 - 207ms/epoch - 13ms/step
Epoch 41/50
16/16 - 0s - loss: 102.8395 - val_loss: 103.5184 - 202ms/epoch - 13ms/step
Epoch 42/50
16/16 - 0s - loss: 103.6218 - val_loss: 103.3720 - 198ms/epoch - 12ms/step
Epoch 43/50
16/16 - 0s - loss: 103.7188 - val_loss: 115.1923 - 201ms/epoch - 13ms/step
Epoch 44/50
16/16 - 0s - loss: 102.4730 - v

Epoch 27/50
16/16 - 0s - loss: 107.2683 - val_loss: 117.0408 - 170ms/epoch - 11ms/step
Epoch 28/50
16/16 - 0s - loss: 107.1858 - val_loss: 113.7114 - 173ms/epoch - 11ms/step
Epoch 29/50
16/16 - 0s - loss: 106.8747 - val_loss: 113.5905 - 172ms/epoch - 11ms/step
Epoch 30/50
16/16 - 0s - loss: 106.9365 - val_loss: 116.1510 - 210ms/epoch - 13ms/step
Epoch 31/50
16/16 - 0s - loss: 107.4104 - val_loss: 115.2268 - 191ms/epoch - 12ms/step
Epoch 32/50
16/16 - 0s - loss: 109.6082 - val_loss: 127.6953 - 222ms/epoch - 14ms/step
Epoch 33/50
16/16 - 0s - loss: 111.3839 - val_loss: 127.3731 - 210ms/epoch - 13ms/step
Epoch 34/50
16/16 - 0s - loss: 110.0827 - val_loss: 113.7010 - 215ms/epoch - 13ms/step
Epoch 35/50
16/16 - 0s - loss: 106.5725 - val_loss: 120.5502 - 193ms/epoch - 12ms/step
Epoch 36/50
16/16 - 0s - loss: 105.9493 - val_loss: 117.7410 - 205ms/epoch - 13ms/step
Epoch 37/50
16/16 - 0s - loss: 108.0675 - val_loss: 113.9548 - 203ms/epoch - 13ms/step
Epoch 38/50
16/16 - 0s - loss: 107.3229 - v

Epoch 22/50
16/16 - 0s - loss: 54.0783 - val_loss: 49.2768 - 218ms/epoch - 14ms/step
Epoch 23/50
16/16 - 0s - loss: 53.1137 - val_loss: 48.7670 - 200ms/epoch - 12ms/step
Epoch 24/50
16/16 - 0s - loss: 54.3416 - val_loss: 57.4277 - 197ms/epoch - 12ms/step
Epoch 25/50
16/16 - 0s - loss: 57.4389 - val_loss: 48.5615 - 193ms/epoch - 12ms/step
Epoch 26/50
16/16 - 0s - loss: 52.8356 - val_loss: 52.4613 - 209ms/epoch - 13ms/step
Epoch 27/50
16/16 - 0s - loss: 52.9533 - val_loss: 48.3281 - 195ms/epoch - 12ms/step
Epoch 28/50
16/16 - 0s - loss: 52.5272 - val_loss: 48.2779 - 201ms/epoch - 13ms/step
Epoch 29/50
16/16 - 0s - loss: 53.7996 - val_loss: 53.6759 - 212ms/epoch - 13ms/step
Epoch 30/50
16/16 - 0s - loss: 57.7595 - val_loss: 48.2087 - 206ms/epoch - 13ms/step
Epoch 31/50
16/16 - 0s - loss: 55.2184 - val_loss: 47.8977 - 200ms/epoch - 12ms/step
Epoch 32/50
16/16 - 0s - loss: 52.5860 - val_loss: 48.2179 - 201ms/epoch - 13ms/step
Epoch 33/50
16/16 - 0s - loss: 52.6785 - val_loss: 47.5453 - 211m

Epoch 18/50
16/16 - 0s - loss: 48.0838 - val_loss: 48.9830 - 204ms/epoch - 13ms/step
Epoch 19/50
16/16 - 0s - loss: 47.5813 - val_loss: 48.8900 - 204ms/epoch - 13ms/step
Epoch 20/50
16/16 - 0s - loss: 51.1833 - val_loss: 59.8206 - 199ms/epoch - 12ms/step
Epoch 21/50
16/16 - 0s - loss: 49.9044 - val_loss: 52.2958 - 208ms/epoch - 13ms/step
Epoch 22/50
16/16 - 0s - loss: 47.2326 - val_loss: 48.9617 - 205ms/epoch - 13ms/step
Epoch 23/50
16/16 - 0s - loss: 47.8190 - val_loss: 48.6155 - 208ms/epoch - 13ms/step
Epoch 24/50
16/16 - 0s - loss: 47.7708 - val_loss: 57.1549 - 200ms/epoch - 12ms/step
Epoch 25/50
16/16 - 0s - loss: 48.9607 - val_loss: 51.2715 - 200ms/epoch - 12ms/step
Epoch 26/50
16/16 - 0s - loss: 50.1222 - val_loss: 51.9026 - 198ms/epoch - 12ms/step
Epoch 27/50
16/16 - 0s - loss: 49.0343 - val_loss: 49.1657 - 226ms/epoch - 14ms/step
Epoch 28/50
16/16 - 0s - loss: 48.4152 - val_loss: 48.4441 - 239ms/epoch - 15ms/step
Epoch 29/50
16/16 - 0s - loss: 48.5930 - val_loss: 50.2967 - 189m

Epoch 14/50
16/16 - 0s - loss: 51.9336 - val_loss: 53.2696 - 196ms/epoch - 12ms/step
Epoch 15/50
16/16 - 0s - loss: 50.3059 - val_loss: 54.3898 - 194ms/epoch - 12ms/step
Epoch 16/50
16/16 - 0s - loss: 51.7302 - val_loss: 52.2429 - 191ms/epoch - 12ms/step
Epoch 17/50
16/16 - 0s - loss: 49.1395 - val_loss: 51.4475 - 196ms/epoch - 12ms/step
Epoch 18/50
16/16 - 0s - loss: 49.3783 - val_loss: 52.4686 - 192ms/epoch - 12ms/step
Epoch 19/50
16/16 - 0s - loss: 46.8810 - val_loss: 51.3582 - 183ms/epoch - 11ms/step
Epoch 20/50
16/16 - 0s - loss: 47.3541 - val_loss: 52.6617 - 197ms/epoch - 12ms/step
Epoch 21/50
16/16 - 0s - loss: 46.8703 - val_loss: 51.0245 - 192ms/epoch - 12ms/step
Epoch 22/50
16/16 - 0s - loss: 47.5517 - val_loss: 50.9212 - 194ms/epoch - 12ms/step
Epoch 23/50
16/16 - 0s - loss: 47.8863 - val_loss: 50.9163 - 190ms/epoch - 12ms/step
Epoch 24/50
16/16 - 0s - loss: 47.0809 - val_loss: 51.1161 - 196ms/epoch - 12ms/step
Epoch 25/50
16/16 - 0s - loss: 46.5525 - val_loss: 50.8833 - 201m

Epoch 10/50
16/16 - 0s - loss: 52.8493 - val_loss: 70.2754 - 190ms/epoch - 12ms/step
Epoch 11/50
16/16 - 0s - loss: 51.8071 - val_loss: 50.3681 - 188ms/epoch - 12ms/step
Epoch 12/50
16/16 - 0s - loss: 47.8083 - val_loss: 51.5568 - 244ms/epoch - 15ms/step
Epoch 13/50
16/16 - 0s - loss: 50.5492 - val_loss: 48.4527 - 193ms/epoch - 12ms/step
Epoch 14/50
16/16 - 0s - loss: 46.7913 - val_loss: 48.4387 - 196ms/epoch - 12ms/step
Epoch 15/50
16/16 - 0s - loss: 46.3099 - val_loss: 48.5761 - 186ms/epoch - 12ms/step
Epoch 16/50
16/16 - 0s - loss: 47.2589 - val_loss: 53.9533 - 190ms/epoch - 12ms/step
Epoch 17/50
16/16 - 0s - loss: 47.0925 - val_loss: 50.8867 - 189ms/epoch - 12ms/step
Epoch 18/50
16/16 - 0s - loss: 46.1678 - val_loss: 48.5769 - 192ms/epoch - 12ms/step
Epoch 19/50
16/16 - 0s - loss: 46.0647 - val_loss: 48.1833 - 190ms/epoch - 12ms/step
Epoch 20/50
16/16 - 0s - loss: 45.8737 - val_loss: 52.2288 - 197ms/epoch - 12ms/step
Epoch 21/50
16/16 - 0s - loss: 47.1070 - val_loss: 48.6383 - 199m

Epoch 6/50
16/16 - 0s - loss: 51.9180 - val_loss: 46.7232 - 226ms/epoch - 14ms/step
Epoch 7/50
16/16 - 0s - loss: 56.3203 - val_loss: 45.1878 - 198ms/epoch - 12ms/step
Epoch 8/50
16/16 - 0s - loss: 54.2184 - val_loss: 64.4780 - 194ms/epoch - 12ms/step
Epoch 9/50
16/16 - 0s - loss: 54.3225 - val_loss: 45.0577 - 195ms/epoch - 12ms/step
Epoch 10/50
16/16 - 0s - loss: 50.0108 - val_loss: 49.6532 - 193ms/epoch - 12ms/step
Epoch 11/50
16/16 - 0s - loss: 52.5597 - val_loss: 49.4317 - 222ms/epoch - 14ms/step
Epoch 12/50
16/16 - 0s - loss: 51.2211 - val_loss: 45.1221 - 199ms/epoch - 12ms/step
Epoch 13/50
16/16 - 0s - loss: 50.5662 - val_loss: 49.3423 - 194ms/epoch - 12ms/step
Epoch 14/50
16/16 - 0s - loss: 53.5535 - val_loss: 43.7088 - 196ms/epoch - 12ms/step
Epoch 15/50
16/16 - 0s - loss: 50.1856 - val_loss: 44.2809 - 194ms/epoch - 12ms/step
Epoch 16/50
16/16 - 0s - loss: 51.2322 - val_loss: 44.4347 - 193ms/epoch - 12ms/step
Epoch 17/50
16/16 - 0s - loss: 50.5408 - val_loss: 48.1371 - 203ms/ep

Epoch 2/50
16/16 - 0s - loss: 47.8163 - val_loss: 46.3144 - 217ms/epoch - 14ms/step
Epoch 3/50
16/16 - 0s - loss: 46.2835 - val_loss: 45.8919 - 201ms/epoch - 13ms/step
Epoch 4/50
16/16 - 0s - loss: 45.9650 - val_loss: 46.5645 - 248ms/epoch - 15ms/step
Epoch 5/50
16/16 - 0s - loss: 46.5062 - val_loss: 48.4497 - 246ms/epoch - 15ms/step
Epoch 6/50
16/16 - 0s - loss: 48.7479 - val_loss: 46.3486 - 196ms/epoch - 12ms/step
Epoch 7/50
16/16 - 0s - loss: 48.7419 - val_loss: 46.1228 - 186ms/epoch - 12ms/step
Epoch 8/50
16/16 - 0s - loss: 46.3070 - val_loss: 49.6494 - 206ms/epoch - 13ms/step
Epoch 9/50
16/16 - 0s - loss: 47.4141 - val_loss: 46.9825 - 161ms/epoch - 10ms/step
Epoch 10/50
16/16 - 0s - loss: 48.1454 - val_loss: 54.1917 - 199ms/epoch - 12ms/step
Epoch 11/50
16/16 - 0s - loss: 48.6433 - val_loss: 46.0694 - 199ms/epoch - 12ms/step
Epoch 12/50
16/16 - 0s - loss: 45.7344 - val_loss: 47.4716 - 204ms/epoch - 13ms/step
Epoch 13/50
16/16 - 0s - loss: 45.3385 - val_loss: 48.3290 - 190ms/epoch 

Epoch 48/50
16/16 - 0s - loss: 45.9029 - val_loss: 44.0242 - 197ms/epoch - 12ms/step
Epoch 49/50
16/16 - 0s - loss: 45.7905 - val_loss: 44.9607 - 229ms/epoch - 14ms/step
Epoch 50/50
16/16 - 0s - loss: 45.8810 - val_loss: 44.6290 - 208ms/epoch - 13ms/step
10/10 [==============================] - 0s 4ms/step
Epoch 1/50
16/16 - 0s - loss: 42.7655 - val_loss: 48.8064 - 358ms/epoch - 22ms/step
Epoch 2/50
16/16 - 0s - loss: 42.1418 - val_loss: 49.1275 - 201ms/epoch - 13ms/step
Epoch 3/50
16/16 - 0s - loss: 41.9530 - val_loss: 50.0563 - 198ms/epoch - 12ms/step
Epoch 4/50
16/16 - 0s - loss: 42.5317 - val_loss: 49.3766 - 197ms/epoch - 12ms/step
Epoch 5/50
16/16 - 0s - loss: 43.1520 - val_loss: 53.6049 - 262ms/epoch - 16ms/step
Epoch 6/50
16/16 - 0s - loss: 44.0470 - val_loss: 49.4887 - 236ms/epoch - 15ms/step
Epoch 7/50
16/16 - 0s - loss: 45.0872 - val_loss: 49.5861 - 248ms/epoch - 15ms/step
Epoch 8/50
16/16 - 0s - loss: 42.3023 - val_loss: 49.9173 - 215ms/epoch - 13ms/step
Epoch 9/50
16/16 - 0

Epoch 44/50
16/16 - 0s - loss: 45.0071 - val_loss: 52.4461 - 233ms/epoch - 15ms/step
Epoch 45/50
16/16 - 0s - loss: 46.3856 - val_loss: 46.8622 - 222ms/epoch - 14ms/step
Epoch 46/50
16/16 - 0s - loss: 44.5624 - val_loss: 47.4381 - 236ms/epoch - 15ms/step
Epoch 47/50
16/16 - 0s - loss: 44.6126 - val_loss: 51.1906 - 196ms/epoch - 12ms/step
Epoch 48/50
16/16 - 0s - loss: 46.8389 - val_loss: 59.1917 - 206ms/epoch - 13ms/step
Epoch 49/50
16/16 - 0s - loss: 50.2530 - val_loss: 54.9325 - 209ms/epoch - 13ms/step
Epoch 50/50
16/16 - 0s - loss: 47.1933 - val_loss: 48.4200 - 225ms/epoch - 14ms/step
10/10 [==============================] - 0s 4ms/step
Epoch 1/50
16/16 - 0s - loss: 46.4508 - val_loss: 44.5879 - 320ms/epoch - 20ms/step
Epoch 2/50
16/16 - 0s - loss: 46.9054 - val_loss: 40.7974 - 204ms/epoch - 13ms/step
Epoch 3/50
16/16 - 0s - loss: 44.1479 - val_loss: 39.8537 - 211ms/epoch - 13ms/step
Epoch 4/50
16/16 - 0s - loss: 42.3020 - val_loss: 40.2736 - 209ms/epoch - 13ms/step
Epoch 5/50
16/16

Epoch 40/50
16/16 - 0s - loss: 44.1920 - val_loss: 43.4058 - 199ms/epoch - 12ms/step
Epoch 41/50
16/16 - 0s - loss: 44.0557 - val_loss: 45.8172 - 200ms/epoch - 12ms/step
Epoch 42/50
16/16 - 0s - loss: 44.1646 - val_loss: 46.8426 - 214ms/epoch - 13ms/step
Epoch 43/50
16/16 - 0s - loss: 44.8931 - val_loss: 45.3756 - 203ms/epoch - 13ms/step
Epoch 44/50
16/16 - 0s - loss: 45.2551 - val_loss: 51.7293 - 207ms/epoch - 13ms/step
Epoch 45/50
16/16 - 0s - loss: 45.5118 - val_loss: 43.9530 - 196ms/epoch - 12ms/step
Epoch 46/50
16/16 - 0s - loss: 43.7405 - val_loss: 43.3036 - 198ms/epoch - 12ms/step
Epoch 47/50
16/16 - 0s - loss: 43.7653 - val_loss: 44.2900 - 204ms/epoch - 13ms/step
Epoch 48/50
16/16 - 0s - loss: 44.5459 - val_loss: 42.7791 - 210ms/epoch - 13ms/step
Epoch 49/50
16/16 - 0s - loss: 43.7266 - val_loss: 43.4285 - 242ms/epoch - 15ms/step
Epoch 50/50
16/16 - 0s - loss: 43.9840 - val_loss: 44.7807 - 207ms/epoch - 13ms/step
10/10 [==============================] - 0s 4ms/step
Epoch 1/50
1

Epoch 36/50
16/16 - 0s - loss: 44.6765 - val_loss: 51.4361 - 202ms/epoch - 13ms/step
Epoch 37/50
16/16 - 0s - loss: 44.7973 - val_loss: 50.8110 - 207ms/epoch - 13ms/step
Epoch 38/50
16/16 - 0s - loss: 45.8612 - val_loss: 51.8269 - 202ms/epoch - 13ms/step
Epoch 39/50
16/16 - 0s - loss: 47.6175 - val_loss: 56.0538 - 217ms/epoch - 14ms/step
Epoch 40/50
16/16 - 0s - loss: 50.2183 - val_loss: 51.0459 - 215ms/epoch - 13ms/step
Epoch 41/50
16/16 - 0s - loss: 45.4935 - val_loss: 50.2197 - 203ms/epoch - 13ms/step
Epoch 42/50
16/16 - 0s - loss: 45.1524 - val_loss: 51.5000 - 215ms/epoch - 13ms/step
Epoch 43/50
16/16 - 0s - loss: 45.3162 - val_loss: 50.6339 - 215ms/epoch - 13ms/step
Epoch 44/50
16/16 - 0s - loss: 46.0499 - val_loss: 54.7506 - 213ms/epoch - 13ms/step
Epoch 45/50
16/16 - 0s - loss: 44.6402 - val_loss: 50.5944 - 210ms/epoch - 13ms/step
Epoch 46/50
16/16 - 0s - loss: 44.2988 - val_loss: 50.3140 - 222ms/epoch - 14ms/step
Epoch 47/50
16/16 - 0s - loss: 44.9631 - val_loss: 50.9414 - 215m

Epoch 32/50
16/16 - 0s - loss: 43.5648 - val_loss: 49.8002 - 196ms/epoch - 12ms/step
Epoch 33/50
16/16 - 0s - loss: 46.5206 - val_loss: 49.9356 - 211ms/epoch - 13ms/step
Epoch 34/50
16/16 - 0s - loss: 48.6695 - val_loss: 64.5735 - 220ms/epoch - 14ms/step
Epoch 35/50
16/16 - 0s - loss: 49.0964 - val_loss: 57.0226 - 205ms/epoch - 13ms/step
Epoch 36/50
16/16 - 0s - loss: 48.6642 - val_loss: 49.6877 - 214ms/epoch - 13ms/step
Epoch 37/50
16/16 - 0s - loss: 43.8935 - val_loss: 49.3818 - 194ms/epoch - 12ms/step
Epoch 38/50
16/16 - 0s - loss: 43.9057 - val_loss: 50.7276 - 217ms/epoch - 14ms/step
Epoch 39/50
16/16 - 0s - loss: 45.1163 - val_loss: 50.8677 - 209ms/epoch - 13ms/step
Epoch 40/50
16/16 - 0s - loss: 43.0213 - val_loss: 49.8923 - 221ms/epoch - 14ms/step
Epoch 41/50
16/16 - 0s - loss: 45.2091 - val_loss: 53.7951 - 253ms/epoch - 16ms/step
Epoch 42/50
16/16 - 0s - loss: 43.2492 - val_loss: 49.4924 - 232ms/epoch - 15ms/step
Epoch 43/50
16/16 - 0s - loss: 43.3500 - val_loss: 49.8049 - 224m

Epoch 28/50
16/16 - 0s - loss: 47.1284 - val_loss: 35.9806 - 229ms/epoch - 14ms/step
Epoch 29/50
16/16 - 0s - loss: 46.6035 - val_loss: 37.3468 - 202ms/epoch - 13ms/step
Epoch 30/50
16/16 - 0s - loss: 46.5996 - val_loss: 40.4157 - 240ms/epoch - 15ms/step
Epoch 31/50
16/16 - 0s - loss: 45.5642 - val_loss: 37.1360 - 218ms/epoch - 14ms/step
Epoch 32/50
16/16 - 0s - loss: 46.0534 - val_loss: 35.5318 - 207ms/epoch - 13ms/step
Epoch 33/50
16/16 - 0s - loss: 48.1405 - val_loss: 35.5662 - 237ms/epoch - 15ms/step
Epoch 34/50
16/16 - 0s - loss: 45.2677 - val_loss: 37.3698 - 218ms/epoch - 14ms/step
Epoch 35/50
16/16 - 0s - loss: 45.5573 - val_loss: 40.4608 - 251ms/epoch - 16ms/step
Epoch 36/50
16/16 - 0s - loss: 48.1739 - val_loss: 39.6381 - 214ms/epoch - 13ms/step
Epoch 37/50
16/16 - 0s - loss: 48.2579 - val_loss: 37.4272 - 227ms/epoch - 14ms/step
Epoch 38/50
16/16 - 0s - loss: 46.8272 - val_loss: 36.7157 - 244ms/epoch - 15ms/step
Epoch 39/50
16/16 - 0s - loss: 46.6177 - val_loss: 35.9334 - 211m

Epoch 24/50
16/16 - 0s - loss: 47.2625 - val_loss: 60.7282 - 214ms/epoch - 13ms/step
Epoch 25/50
16/16 - 0s - loss: 48.1605 - val_loss: 51.7318 - 251ms/epoch - 16ms/step
Epoch 26/50
16/16 - 0s - loss: 47.4584 - val_loss: 51.8912 - 221ms/epoch - 14ms/step
Epoch 27/50
16/16 - 0s - loss: 46.2311 - val_loss: 51.8907 - 198ms/epoch - 12ms/step
Epoch 28/50
16/16 - 0s - loss: 46.5570 - val_loss: 52.2293 - 225ms/epoch - 14ms/step
Epoch 29/50
16/16 - 0s - loss: 46.5084 - val_loss: 52.5902 - 234ms/epoch - 15ms/step
Epoch 30/50
16/16 - 0s - loss: 47.3881 - val_loss: 54.1345 - 216ms/epoch - 13ms/step
Epoch 31/50
16/16 - 0s - loss: 47.4847 - val_loss: 54.4366 - 206ms/epoch - 13ms/step
Epoch 32/50
16/16 - 0s - loss: 48.2561 - val_loss: 52.7421 - 213ms/epoch - 13ms/step
Epoch 33/50
16/16 - 0s - loss: 47.0880 - val_loss: 51.8100 - 199ms/epoch - 12ms/step
Epoch 34/50
16/16 - 0s - loss: 45.7978 - val_loss: 52.1749 - 216ms/epoch - 14ms/step
Epoch 35/50
16/16 - 0s - loss: 46.2029 - val_loss: 52.5365 - 219m

Epoch 20/50
16/16 - 0s - loss: 41.5997 - val_loss: 45.7699 - 207ms/epoch - 13ms/step
Epoch 21/50
16/16 - 0s - loss: 46.7767 - val_loss: 46.5186 - 209ms/epoch - 13ms/step
Epoch 22/50
16/16 - 0s - loss: 41.5665 - val_loss: 46.3090 - 213ms/epoch - 13ms/step
Epoch 23/50
16/16 - 0s - loss: 43.0021 - val_loss: 46.2107 - 204ms/epoch - 13ms/step
Epoch 24/50
16/16 - 0s - loss: 43.1246 - val_loss: 45.4993 - 207ms/epoch - 13ms/step
Epoch 25/50
16/16 - 0s - loss: 41.6680 - val_loss: 45.2736 - 212ms/epoch - 13ms/step
Epoch 26/50
16/16 - 0s - loss: 41.6414 - val_loss: 45.8883 - 217ms/epoch - 14ms/step
Epoch 27/50
16/16 - 0s - loss: 41.5916 - val_loss: 45.8226 - 221ms/epoch - 14ms/step
Epoch 28/50
16/16 - 0s - loss: 43.7763 - val_loss: 50.2397 - 209ms/epoch - 13ms/step
Epoch 29/50
16/16 - 0s - loss: 45.4375 - val_loss: 48.8926 - 222ms/epoch - 14ms/step
Epoch 30/50
16/16 - 0s - loss: 42.3167 - val_loss: 49.4687 - 203ms/epoch - 13ms/step
Epoch 31/50
16/16 - 0s - loss: 42.7148 - val_loss: 47.3672 - 218m

## Calculating mean and standard deviation of the MSEs using numpy 

In [22]:
mse_arr = np.array(mse_50)
mean_mse = np.mean(mse_arr)
std_mse = np.std(mse_arr)

In [23]:
print('The mean MSE after training the model 50 times is: ',mean_mse, ' and the standard deviation is: ', std_mse)

The mean MSE after training the model 50 times is:  92.61864332056368  and the standard deviation is:  85.06794677610809


In part A the mean MSE and standard deviation was 58.269382589504986 and 18.803301390418138 respectively. 
Training the model on a normalized data in this particular dataset has led to increase in the error and the resulting error and standard deviation in MSE is 92.61864332056368  and 85.06794677610809 respectively

## using the MinMax scaler to scale the data instead of subtracting the mean and dividing by the standard deviation

In [25]:
# Splitting the raw data into training and testing and using the minmax scaler instance to fit and transform the training data
#and using the same transform on the test data:

mse_50_Minmax = []
minmax = MinMaxScaler()
model = regression_model()
for i in range(50):
    X_train, X_test, y_train, y_test = train_test_split(predictors, target, test_size=0.30)
    X_train_scaled = minmax.fit_transform(X_train)
    X_test_scaled = minmax.transform(X_test)
    model.fit(X_train_scaled, y_train, validation_split=0.3, epochs=50, verbose=2)
    predictions = model.predict(X_test_scaled)
    mse_50_Minmax.append(mean_squared_error(y_test, predictions))






Epoch 1/50
16/16 - 1s - loss: 1634.5219 - val_loss: 1482.4193 - 1s/epoch - 94ms/step
Epoch 2/50
16/16 - 0s - loss: 1622.8274 - val_loss: 1470.9933 - 152ms/epoch - 9ms/step
Epoch 3/50
16/16 - 0s - loss: 1610.4081 - val_loss: 1458.5173 - 167ms/epoch - 10ms/step
Epoch 4/50
16/16 - 0s - loss: 1596.8229 - val_loss: 1444.5411 - 173ms/epoch - 11ms/step
Epoch 5/50
16/16 - 0s - loss: 1581.3524 - val_loss: 1428.1656 - 171ms/epoch - 11ms/step
Epoch 6/50
16/16 - 0s - loss: 1562.6427 - val_loss: 1408.6860 - 151ms/epoch - 9ms/step
Epoch 7/50
16/16 - 0s - loss: 1540.6851 - val_loss: 1386.1021 - 145ms/epoch - 9ms/step
Epoch 8/50
16/16 - 0s - loss: 1516.1047 - val_loss: 1361.8228 - 128ms/epoch - 8ms/step
Epoch 9/50
16/16 - 0s - loss: 1489.8262 - val_loss: 1336.4254 - 136ms/epoch - 9ms/step
Epoch 10/50
16/16 - 0s - loss: 1462.3156 - val_loss: 1309.1345 - 157ms/epoch - 10ms/step
Epoch 11/50
16/16 - 0s - loss: 1433.1158 - val_loss: 1280.8318 - 195ms/epoch - 12ms/step
Epoch 12/50
16/16 - 0s - loss: 1402.48

Epoch 45/50
16/16 - 0s - loss: 224.5410 - val_loss: 210.8817 - 259ms/epoch - 16ms/step
Epoch 46/50
16/16 - 0s - loss: 223.7206 - val_loss: 210.1602 - 254ms/epoch - 16ms/step
Epoch 47/50
16/16 - 0s - loss: 222.8846 - val_loss: 209.4972 - 297ms/epoch - 19ms/step
Epoch 48/50
16/16 - 0s - loss: 222.0203 - val_loss: 208.8576 - 236ms/epoch - 15ms/step
Epoch 49/50
16/16 - 0s - loss: 221.1788 - val_loss: 207.9848 - 231ms/epoch - 14ms/step
Epoch 50/50
16/16 - 0s - loss: 220.3149 - val_loss: 207.2804 - 229ms/epoch - 14ms/step
10/10 [==============================] - 0s 4ms/step
Epoch 1/50
16/16 - 0s - loss: 200.7360 - val_loss: 249.3483 - 319ms/epoch - 20ms/step
Epoch 2/50
16/16 - 0s - loss: 199.8020 - val_loss: 248.9186 - 223ms/epoch - 14ms/step
Epoch 3/50
16/16 - 0s - loss: 198.6986 - val_loss: 248.2547 - 262ms/epoch - 16ms/step
Epoch 4/50
16/16 - 0s - loss: 197.7593 - val_loss: 247.7529 - 254ms/epoch - 16ms/step
Epoch 5/50
16/16 - 0s - loss: 196.7939 - val_loss: 247.1456 - 238ms/epoch - 15ms/

Epoch 39/50
16/16 - 0s - loss: 145.1221 - val_loss: 141.4131 - 199ms/epoch - 12ms/step
Epoch 40/50
16/16 - 0s - loss: 144.3670 - val_loss: 140.8482 - 195ms/epoch - 12ms/step
Epoch 41/50
16/16 - 0s - loss: 143.6322 - val_loss: 140.2848 - 177ms/epoch - 11ms/step
Epoch 42/50
16/16 - 0s - loss: 142.8885 - val_loss: 139.7439 - 200ms/epoch - 13ms/step
Epoch 43/50
16/16 - 0s - loss: 142.1926 - val_loss: 139.1767 - 233ms/epoch - 15ms/step
Epoch 44/50
16/16 - 0s - loss: 141.4954 - val_loss: 138.5971 - 246ms/epoch - 15ms/step
Epoch 45/50
16/16 - 0s - loss: 140.7646 - val_loss: 138.0604 - 230ms/epoch - 14ms/step
Epoch 46/50
16/16 - 0s - loss: 140.0488 - val_loss: 137.5679 - 209ms/epoch - 13ms/step
Epoch 47/50
16/16 - 0s - loss: 139.4311 - val_loss: 137.0069 - 220ms/epoch - 14ms/step
Epoch 48/50
16/16 - 0s - loss: 138.7744 - val_loss: 136.5235 - 254ms/epoch - 16ms/step
Epoch 49/50
16/16 - 0s - loss: 138.0562 - val_loss: 135.9951 - 211ms/epoch - 13ms/step
Epoch 50/50
16/16 - 0s - loss: 137.4102 - v

Epoch 33/50
16/16 - 0s - loss: 110.1402 - val_loss: 110.6626 - 260ms/epoch - 16ms/step
Epoch 34/50
16/16 - 0s - loss: 109.9599 - val_loss: 110.2945 - 238ms/epoch - 15ms/step
Epoch 35/50
16/16 - 0s - loss: 109.7777 - val_loss: 110.2929 - 232ms/epoch - 15ms/step
Epoch 36/50
16/16 - 0s - loss: 109.6102 - val_loss: 110.3621 - 203ms/epoch - 13ms/step
Epoch 37/50
16/16 - 0s - loss: 109.4036 - val_loss: 110.1880 - 203ms/epoch - 13ms/step
Epoch 38/50
16/16 - 0s - loss: 109.1978 - val_loss: 109.8687 - 206ms/epoch - 13ms/step
Epoch 39/50
16/16 - 0s - loss: 109.0479 - val_loss: 109.5881 - 243ms/epoch - 15ms/step
Epoch 40/50
16/16 - 0s - loss: 108.8591 - val_loss: 109.6032 - 235ms/epoch - 15ms/step
Epoch 41/50
16/16 - 0s - loss: 108.6903 - val_loss: 109.7286 - 221ms/epoch - 14ms/step
Epoch 42/50
16/16 - 0s - loss: 108.5197 - val_loss: 109.4421 - 232ms/epoch - 15ms/step
Epoch 43/50
16/16 - 0s - loss: 108.3416 - val_loss: 109.3366 - 211ms/epoch - 13ms/step
Epoch 44/50
16/16 - 0s - loss: 108.2011 - v

Epoch 27/50
16/16 - 0s - loss: 107.9673 - val_loss: 93.0356 - 216ms/epoch - 13ms/step
Epoch 28/50
16/16 - 0s - loss: 107.8399 - val_loss: 92.9765 - 220ms/epoch - 14ms/step
Epoch 29/50
16/16 - 0s - loss: 107.7398 - val_loss: 93.0065 - 201ms/epoch - 13ms/step
Epoch 30/50
16/16 - 0s - loss: 107.6301 - val_loss: 92.9592 - 213ms/epoch - 13ms/step
Epoch 31/50
16/16 - 0s - loss: 107.5405 - val_loss: 92.9272 - 228ms/epoch - 14ms/step
Epoch 32/50
16/16 - 0s - loss: 107.4233 - val_loss: 92.9573 - 235ms/epoch - 15ms/step
Epoch 33/50
16/16 - 0s - loss: 107.3352 - val_loss: 92.9750 - 208ms/epoch - 13ms/step
Epoch 34/50
16/16 - 0s - loss: 107.3661 - val_loss: 92.8209 - 228ms/epoch - 14ms/step
Epoch 35/50
16/16 - 0s - loss: 107.0835 - val_loss: 92.8652 - 224ms/epoch - 14ms/step
Epoch 36/50
16/16 - 0s - loss: 106.9761 - val_loss: 92.8834 - 193ms/epoch - 12ms/step
Epoch 37/50
16/16 - 0s - loss: 106.9546 - val_loss: 92.9812 - 197ms/epoch - 12ms/step
Epoch 38/50
16/16 - 0s - loss: 106.7930 - val_loss: 92

Epoch 22/50
16/16 - 0s - loss: 87.2251 - val_loss: 101.0282 - 230ms/epoch - 14ms/step
Epoch 23/50
16/16 - 0s - loss: 87.0562 - val_loss: 100.8452 - 227ms/epoch - 14ms/step
Epoch 24/50
16/16 - 0s - loss: 86.8712 - val_loss: 100.5014 - 216ms/epoch - 14ms/step
Epoch 25/50
16/16 - 0s - loss: 86.8236 - val_loss: 100.0637 - 216ms/epoch - 14ms/step
Epoch 26/50
16/16 - 0s - loss: 86.7903 - val_loss: 99.8831 - 221ms/epoch - 14ms/step
Epoch 27/50
16/16 - 0s - loss: 86.6105 - val_loss: 100.2036 - 245ms/epoch - 15ms/step
Epoch 28/50
16/16 - 0s - loss: 86.5429 - val_loss: 100.4146 - 212ms/epoch - 13ms/step
Epoch 29/50
16/16 - 0s - loss: 86.4509 - val_loss: 100.3453 - 217ms/epoch - 14ms/step
Epoch 30/50
16/16 - 0s - loss: 86.3694 - val_loss: 100.1282 - 202ms/epoch - 13ms/step
Epoch 31/50
16/16 - 0s - loss: 86.2906 - val_loss: 99.9521 - 205ms/epoch - 13ms/step
Epoch 32/50
16/16 - 0s - loss: 86.1853 - val_loss: 100.0780 - 211ms/epoch - 13ms/step
Epoch 33/50
16/16 - 0s - loss: 86.1051 - val_loss: 100.0

Epoch 18/50
16/16 - 0s - loss: 90.8619 - val_loss: 92.8027 - 199ms/epoch - 12ms/step
Epoch 19/50
16/16 - 0s - loss: 90.6535 - val_loss: 92.9117 - 205ms/epoch - 13ms/step
Epoch 20/50
16/16 - 0s - loss: 90.5687 - val_loss: 93.0493 - 222ms/epoch - 14ms/step
Epoch 21/50
16/16 - 0s - loss: 90.4743 - val_loss: 93.0724 - 211ms/epoch - 13ms/step
Epoch 22/50
16/16 - 0s - loss: 90.3946 - val_loss: 92.8510 - 225ms/epoch - 14ms/step
Epoch 23/50
16/16 - 0s - loss: 90.2983 - val_loss: 92.7823 - 228ms/epoch - 14ms/step
Epoch 24/50
16/16 - 0s - loss: 90.1547 - val_loss: 92.9157 - 215ms/epoch - 13ms/step
Epoch 25/50
16/16 - 0s - loss: 90.1629 - val_loss: 92.9888 - 218ms/epoch - 14ms/step
Epoch 26/50
16/16 - 0s - loss: 90.0341 - val_loss: 92.9987 - 222ms/epoch - 14ms/step
Epoch 27/50
16/16 - 0s - loss: 89.9110 - val_loss: 92.6663 - 202ms/epoch - 13ms/step
Epoch 28/50
16/16 - 0s - loss: 89.7969 - val_loss: 92.5415 - 206ms/epoch - 13ms/step
Epoch 29/50
16/16 - 0s - loss: 89.6995 - val_loss: 92.4667 - 211m

Epoch 14/50
16/16 - 0s - loss: 80.0246 - val_loss: 75.4227 - 215ms/epoch - 13ms/step
Epoch 15/50
16/16 - 0s - loss: 80.0570 - val_loss: 75.5534 - 204ms/epoch - 13ms/step
Epoch 16/50
16/16 - 0s - loss: 79.8527 - val_loss: 75.4091 - 205ms/epoch - 13ms/step
Epoch 17/50
16/16 - 0s - loss: 79.7510 - val_loss: 75.3373 - 208ms/epoch - 13ms/step
Epoch 18/50
16/16 - 0s - loss: 79.6514 - val_loss: 75.4769 - 237ms/epoch - 15ms/step
Epoch 19/50
16/16 - 0s - loss: 79.5322 - val_loss: 75.3423 - 254ms/epoch - 16ms/step
Epoch 20/50
16/16 - 0s - loss: 79.4242 - val_loss: 75.3002 - 222ms/epoch - 14ms/step
Epoch 21/50
16/16 - 0s - loss: 79.3199 - val_loss: 75.3547 - 231ms/epoch - 14ms/step
Epoch 22/50
16/16 - 0s - loss: 79.2703 - val_loss: 75.4551 - 237ms/epoch - 15ms/step
Epoch 23/50
16/16 - 0s - loss: 79.1307 - val_loss: 75.3667 - 243ms/epoch - 15ms/step
Epoch 24/50
16/16 - 0s - loss: 79.1207 - val_loss: 75.1481 - 252ms/epoch - 16ms/step
Epoch 25/50
16/16 - 0s - loss: 78.9807 - val_loss: 75.2923 - 245m

Epoch 10/50
16/16 - 0s - loss: 73.5785 - val_loss: 82.7134 - 227ms/epoch - 14ms/step
Epoch 11/50
16/16 - 0s - loss: 73.5851 - val_loss: 82.6238 - 229ms/epoch - 14ms/step
Epoch 12/50
16/16 - 0s - loss: 73.5060 - val_loss: 82.5363 - 217ms/epoch - 14ms/step
Epoch 13/50
16/16 - 0s - loss: 73.4370 - val_loss: 82.4694 - 225ms/epoch - 14ms/step
Epoch 14/50
16/16 - 0s - loss: 73.2695 - val_loss: 82.3641 - 213ms/epoch - 13ms/step
Epoch 15/50
16/16 - 0s - loss: 73.1843 - val_loss: 82.2867 - 232ms/epoch - 14ms/step
Epoch 16/50
16/16 - 0s - loss: 73.1454 - val_loss: 82.2022 - 217ms/epoch - 14ms/step
Epoch 17/50
16/16 - 0s - loss: 73.0374 - val_loss: 82.1359 - 215ms/epoch - 13ms/step
Epoch 18/50
16/16 - 0s - loss: 72.9893 - val_loss: 82.0666 - 221ms/epoch - 14ms/step
Epoch 19/50
16/16 - 0s - loss: 72.9108 - val_loss: 82.0464 - 219ms/epoch - 14ms/step
Epoch 20/50
16/16 - 0s - loss: 72.8278 - val_loss: 81.9358 - 230ms/epoch - 14ms/step
Epoch 21/50
16/16 - 0s - loss: 72.8704 - val_loss: 81.9017 - 271m

Epoch 6/50
16/16 - 0s - loss: 67.9528 - val_loss: 86.2510 - 229ms/epoch - 14ms/step
Epoch 7/50
16/16 - 0s - loss: 67.7199 - val_loss: 85.9977 - 222ms/epoch - 14ms/step
Epoch 8/50
16/16 - 0s - loss: 67.6339 - val_loss: 85.8901 - 242ms/epoch - 15ms/step
Epoch 9/50
16/16 - 0s - loss: 67.5434 - val_loss: 85.7537 - 216ms/epoch - 14ms/step
Epoch 10/50
16/16 - 0s - loss: 67.4921 - val_loss: 85.6418 - 217ms/epoch - 14ms/step
Epoch 11/50
16/16 - 0s - loss: 67.4331 - val_loss: 85.7027 - 216ms/epoch - 13ms/step
Epoch 12/50
16/16 - 0s - loss: 67.2822 - val_loss: 85.6114 - 213ms/epoch - 13ms/step
Epoch 13/50
16/16 - 0s - loss: 67.2247 - val_loss: 85.4310 - 231ms/epoch - 14ms/step
Epoch 14/50
16/16 - 0s - loss: 67.1466 - val_loss: 85.4084 - 216ms/epoch - 13ms/step
Epoch 15/50
16/16 - 0s - loss: 67.0635 - val_loss: 85.2935 - 210ms/epoch - 13ms/step
Epoch 16/50
16/16 - 0s - loss: 66.9962 - val_loss: 85.1766 - 214ms/epoch - 13ms/step
Epoch 17/50
16/16 - 0s - loss: 66.9372 - val_loss: 85.1467 - 219ms/ep

Epoch 2/50
16/16 - 0s - loss: 58.2438 - val_loss: 76.9294 - 233ms/epoch - 15ms/step
Epoch 3/50
16/16 - 0s - loss: 58.0822 - val_loss: 76.5735 - 214ms/epoch - 13ms/step
Epoch 4/50
16/16 - 0s - loss: 58.0454 - val_loss: 76.4021 - 223ms/epoch - 14ms/step
Epoch 5/50
16/16 - 0s - loss: 57.9206 - val_loss: 76.7936 - 220ms/epoch - 14ms/step
Epoch 6/50
16/16 - 0s - loss: 57.9131 - val_loss: 76.7770 - 211ms/epoch - 13ms/step
Epoch 7/50
16/16 - 0s - loss: 57.8241 - val_loss: 76.7899 - 229ms/epoch - 14ms/step
Epoch 8/50
16/16 - 0s - loss: 57.7621 - val_loss: 76.5455 - 227ms/epoch - 14ms/step
Epoch 9/50
16/16 - 0s - loss: 57.7270 - val_loss: 76.8844 - 220ms/epoch - 14ms/step
Epoch 10/50
16/16 - 0s - loss: 57.6609 - val_loss: 76.5346 - 225ms/epoch - 14ms/step
Epoch 11/50
16/16 - 0s - loss: 57.5738 - val_loss: 76.3080 - 222ms/epoch - 14ms/step
Epoch 12/50
16/16 - 0s - loss: 57.5041 - val_loss: 76.3802 - 242ms/epoch - 15ms/step
Epoch 13/50
16/16 - 0s - loss: 57.4375 - val_loss: 76.4810 - 258ms/epoch 

Epoch 48/50
16/16 - 0s - loss: 58.6288 - val_loss: 62.4209 - 231ms/epoch - 14ms/step
Epoch 49/50
16/16 - 0s - loss: 58.6647 - val_loss: 62.5706 - 219ms/epoch - 14ms/step
Epoch 50/50
16/16 - 0s - loss: 58.5485 - val_loss: 62.3412 - 216ms/epoch - 14ms/step
10/10 [==============================] - 0s 4ms/step
Epoch 1/50
16/16 - 0s - loss: 59.8496 - val_loss: 66.1293 - 316ms/epoch - 20ms/step
Epoch 2/50
16/16 - 0s - loss: 59.3282 - val_loss: 66.5774 - 217ms/epoch - 14ms/step
Epoch 3/50
16/16 - 0s - loss: 59.1850 - val_loss: 67.0574 - 222ms/epoch - 14ms/step
Epoch 4/50
16/16 - 0s - loss: 59.1065 - val_loss: 67.2934 - 214ms/epoch - 13ms/step
Epoch 5/50
16/16 - 0s - loss: 58.8776 - val_loss: 66.7484 - 216ms/epoch - 14ms/step
Epoch 6/50
16/16 - 0s - loss: 58.8412 - val_loss: 66.9329 - 232ms/epoch - 15ms/step
Epoch 7/50
16/16 - 0s - loss: 58.6893 - val_loss: 66.6716 - 224ms/epoch - 14ms/step
Epoch 8/50
16/16 - 0s - loss: 58.5644 - val_loss: 66.7982 - 217ms/epoch - 14ms/step
Epoch 9/50
16/16 - 0

Epoch 44/50
16/16 - 0s - loss: 53.3185 - val_loss: 54.5107 - 230ms/epoch - 14ms/step
Epoch 45/50
16/16 - 0s - loss: 53.2880 - val_loss: 54.4747 - 212ms/epoch - 13ms/step
Epoch 46/50
16/16 - 0s - loss: 53.2621 - val_loss: 54.4769 - 199ms/epoch - 12ms/step
Epoch 47/50
16/16 - 0s - loss: 53.2122 - val_loss: 54.4547 - 211ms/epoch - 13ms/step
Epoch 48/50
16/16 - 0s - loss: 53.1776 - val_loss: 54.4271 - 242ms/epoch - 15ms/step
Epoch 49/50
16/16 - 0s - loss: 53.1828 - val_loss: 54.3944 - 276ms/epoch - 17ms/step
Epoch 50/50
16/16 - 0s - loss: 53.1216 - val_loss: 54.4162 - 208ms/epoch - 13ms/step
10/10 [==============================] - 0s 4ms/step
Epoch 1/50
16/16 - 0s - loss: 57.0737 - val_loss: 57.5062 - 326ms/epoch - 20ms/step
Epoch 2/50
16/16 - 0s - loss: 56.9894 - val_loss: 57.4959 - 217ms/epoch - 14ms/step
Epoch 3/50
16/16 - 0s - loss: 56.9196 - val_loss: 57.5391 - 195ms/epoch - 12ms/step
Epoch 4/50
16/16 - 0s - loss: 56.8941 - val_loss: 57.5437 - 209ms/epoch - 13ms/step
Epoch 5/50
16/16

Epoch 40/50
16/16 - 0s - loss: 54.4581 - val_loss: 54.4708 - 252ms/epoch - 16ms/step
Epoch 41/50
16/16 - 0s - loss: 54.3567 - val_loss: 54.4970 - 243ms/epoch - 15ms/step
Epoch 42/50
16/16 - 0s - loss: 54.3555 - val_loss: 54.5998 - 263ms/epoch - 16ms/step
Epoch 43/50
16/16 - 0s - loss: 54.4175 - val_loss: 54.4649 - 265ms/epoch - 17ms/step
Epoch 44/50
16/16 - 0s - loss: 54.2560 - val_loss: 54.5118 - 253ms/epoch - 16ms/step
Epoch 45/50
16/16 - 0s - loss: 54.3314 - val_loss: 54.4852 - 262ms/epoch - 16ms/step
Epoch 46/50
16/16 - 0s - loss: 54.1900 - val_loss: 54.5669 - 250ms/epoch - 16ms/step
Epoch 47/50
16/16 - 0s - loss: 54.1826 - val_loss: 54.5480 - 262ms/epoch - 16ms/step
Epoch 48/50
16/16 - 0s - loss: 54.1521 - val_loss: 54.4372 - 246ms/epoch - 15ms/step
Epoch 49/50
16/16 - 0s - loss: 54.2734 - val_loss: 54.3579 - 232ms/epoch - 14ms/step
Epoch 50/50
16/16 - 0s - loss: 54.0223 - val_loss: 54.4931 - 241ms/epoch - 15ms/step
10/10 [==============================] - 0s 4ms/step
Epoch 1/50
1

Epoch 36/50
16/16 - 0s - loss: 52.0911 - val_loss: 53.9365 - 222ms/epoch - 14ms/step
Epoch 37/50
16/16 - 0s - loss: 52.1358 - val_loss: 54.2020 - 238ms/epoch - 15ms/step
Epoch 38/50
16/16 - 0s - loss: 52.0786 - val_loss: 53.9688 - 240ms/epoch - 15ms/step
Epoch 39/50
16/16 - 0s - loss: 52.1287 - val_loss: 53.9142 - 275ms/epoch - 17ms/step
Epoch 40/50
16/16 - 0s - loss: 52.0171 - val_loss: 54.1744 - 245ms/epoch - 15ms/step
Epoch 41/50
16/16 - 0s - loss: 51.9874 - val_loss: 54.1454 - 254ms/epoch - 16ms/step
Epoch 42/50
16/16 - 0s - loss: 51.9584 - val_loss: 54.0387 - 242ms/epoch - 15ms/step
Epoch 43/50
16/16 - 0s - loss: 51.9759 - val_loss: 53.8855 - 268ms/epoch - 17ms/step
Epoch 44/50
16/16 - 0s - loss: 51.9440 - val_loss: 54.0264 - 245ms/epoch - 15ms/step
Epoch 45/50
16/16 - 0s - loss: 51.9447 - val_loss: 53.9981 - 262ms/epoch - 16ms/step
Epoch 46/50
16/16 - 0s - loss: 52.0164 - val_loss: 54.1410 - 254ms/epoch - 16ms/step
Epoch 47/50
16/16 - 0s - loss: 51.9056 - val_loss: 53.9907 - 242m

Epoch 32/50
16/16 - 0s - loss: 48.9981 - val_loss: 59.7526 - 251ms/epoch - 16ms/step
Epoch 33/50
16/16 - 0s - loss: 49.0344 - val_loss: 59.7363 - 232ms/epoch - 15ms/step
Epoch 34/50
16/16 - 0s - loss: 48.9890 - val_loss: 59.7457 - 233ms/epoch - 15ms/step
Epoch 35/50
16/16 - 0s - loss: 49.0110 - val_loss: 59.8614 - 226ms/epoch - 14ms/step
Epoch 36/50
16/16 - 0s - loss: 48.9543 - val_loss: 59.8023 - 218ms/epoch - 14ms/step
Epoch 37/50
16/16 - 0s - loss: 48.9446 - val_loss: 59.8900 - 246ms/epoch - 15ms/step
Epoch 38/50
16/16 - 0s - loss: 48.8461 - val_loss: 59.7876 - 224ms/epoch - 14ms/step
Epoch 39/50
16/16 - 0s - loss: 48.8236 - val_loss: 59.7753 - 233ms/epoch - 15ms/step
Epoch 40/50
16/16 - 0s - loss: 48.8623 - val_loss: 59.7844 - 240ms/epoch - 15ms/step
Epoch 41/50
16/16 - 0s - loss: 48.8402 - val_loss: 59.9579 - 216ms/epoch - 14ms/step
Epoch 42/50
16/16 - 0s - loss: 48.8229 - val_loss: 59.8478 - 207ms/epoch - 13ms/step
Epoch 43/50
16/16 - 0s - loss: 48.7587 - val_loss: 59.8845 - 200m

Epoch 28/50
16/16 - 0s - loss: 44.5692 - val_loss: 51.8180 - 223ms/epoch - 14ms/step
Epoch 29/50
16/16 - 0s - loss: 44.5652 - val_loss: 51.7672 - 219ms/epoch - 14ms/step
Epoch 30/50
16/16 - 0s - loss: 44.5242 - val_loss: 51.7905 - 225ms/epoch - 14ms/step
Epoch 31/50
16/16 - 0s - loss: 44.5332 - val_loss: 51.7973 - 216ms/epoch - 13ms/step
Epoch 32/50
16/16 - 0s - loss: 44.5882 - val_loss: 51.8615 - 220ms/epoch - 14ms/step
Epoch 33/50
16/16 - 0s - loss: 44.4905 - val_loss: 51.7588 - 226ms/epoch - 14ms/step
Epoch 34/50
16/16 - 0s - loss: 44.5249 - val_loss: 51.7664 - 240ms/epoch - 15ms/step
Epoch 35/50
16/16 - 0s - loss: 44.4726 - val_loss: 51.7770 - 238ms/epoch - 15ms/step
Epoch 36/50
16/16 - 0s - loss: 44.4700 - val_loss: 51.8383 - 235ms/epoch - 15ms/step
Epoch 37/50
16/16 - 0s - loss: 44.4696 - val_loss: 51.8139 - 218ms/epoch - 14ms/step
Epoch 38/50
16/16 - 0s - loss: 44.4702 - val_loss: 51.8863 - 219ms/epoch - 14ms/step
Epoch 39/50
16/16 - 0s - loss: 44.4423 - val_loss: 51.7573 - 222m

Epoch 24/50
16/16 - 0s - loss: 50.5707 - val_loss: 49.8750 - 234ms/epoch - 15ms/step
Epoch 25/50
16/16 - 0s - loss: 50.5551 - val_loss: 49.9351 - 230ms/epoch - 14ms/step
Epoch 26/50
16/16 - 0s - loss: 50.5036 - val_loss: 49.8921 - 262ms/epoch - 16ms/step
Epoch 27/50
16/16 - 0s - loss: 50.5367 - val_loss: 49.9548 - 241ms/epoch - 15ms/step
Epoch 28/50
16/16 - 0s - loss: 50.4843 - val_loss: 49.9163 - 237ms/epoch - 15ms/step
Epoch 29/50
16/16 - 0s - loss: 50.4259 - val_loss: 49.8471 - 224ms/epoch - 14ms/step
Epoch 30/50
16/16 - 0s - loss: 50.4301 - val_loss: 49.8132 - 213ms/epoch - 13ms/step
Epoch 31/50
16/16 - 0s - loss: 50.4377 - val_loss: 49.8742 - 214ms/epoch - 13ms/step
Epoch 32/50
16/16 - 0s - loss: 50.3994 - val_loss: 49.8257 - 217ms/epoch - 14ms/step
Epoch 33/50
16/16 - 0s - loss: 50.3283 - val_loss: 49.8438 - 222ms/epoch - 14ms/step
Epoch 34/50
16/16 - 0s - loss: 50.3182 - val_loss: 49.7928 - 232ms/epoch - 14ms/step
Epoch 35/50
16/16 - 0s - loss: 50.3299 - val_loss: 49.8274 - 224m

Epoch 20/50
16/16 - 0s - loss: 48.2930 - val_loss: 43.1629 - 292ms/epoch - 18ms/step
Epoch 21/50
16/16 - 0s - loss: 48.2539 - val_loss: 43.1556 - 230ms/epoch - 14ms/step
Epoch 22/50
16/16 - 0s - loss: 48.2118 - val_loss: 43.1611 - 257ms/epoch - 16ms/step
Epoch 23/50
16/16 - 0s - loss: 48.1869 - val_loss: 43.1441 - 242ms/epoch - 15ms/step
Epoch 24/50
16/16 - 0s - loss: 48.1668 - val_loss: 43.1490 - 249ms/epoch - 16ms/step
Epoch 25/50
16/16 - 0s - loss: 48.1602 - val_loss: 43.1819 - 259ms/epoch - 16ms/step
Epoch 26/50
16/16 - 0s - loss: 48.1487 - val_loss: 43.1898 - 235ms/epoch - 15ms/step
Epoch 27/50
16/16 - 0s - loss: 48.2506 - val_loss: 43.2532 - 286ms/epoch - 18ms/step
Epoch 28/50
16/16 - 0s - loss: 48.0842 - val_loss: 43.2075 - 304ms/epoch - 19ms/step
Epoch 29/50
16/16 - 0s - loss: 48.0628 - val_loss: 43.1980 - 251ms/epoch - 16ms/step
Epoch 30/50
16/16 - 0s - loss: 48.0088 - val_loss: 43.2139 - 244ms/epoch - 15ms/step
Epoch 31/50
16/16 - 0s - loss: 48.0656 - val_loss: 43.3051 - 269m

Epoch 16/50
16/16 - 0s - loss: 41.3782 - val_loss: 43.6862 - 246ms/epoch - 15ms/step
Epoch 17/50
16/16 - 0s - loss: 41.3788 - val_loss: 44.0467 - 245ms/epoch - 15ms/step
Epoch 18/50
16/16 - 0s - loss: 41.3287 - val_loss: 43.8944 - 246ms/epoch - 15ms/step
Epoch 19/50
16/16 - 0s - loss: 41.3028 - val_loss: 43.8332 - 231ms/epoch - 14ms/step
Epoch 20/50
16/16 - 0s - loss: 41.2921 - val_loss: 43.7500 - 219ms/epoch - 14ms/step
Epoch 21/50
16/16 - 0s - loss: 41.2749 - val_loss: 44.0068 - 256ms/epoch - 16ms/step
Epoch 22/50
16/16 - 0s - loss: 41.2367 - val_loss: 43.7962 - 224ms/epoch - 14ms/step
Epoch 23/50
16/16 - 0s - loss: 41.2057 - val_loss: 43.7577 - 220ms/epoch - 14ms/step
Epoch 24/50
16/16 - 0s - loss: 41.1743 - val_loss: 43.7414 - 220ms/epoch - 14ms/step
Epoch 25/50
16/16 - 0s - loss: 41.1407 - val_loss: 43.8317 - 236ms/epoch - 15ms/step
Epoch 26/50
16/16 - 0s - loss: 41.1569 - val_loss: 43.9317 - 240ms/epoch - 15ms/step
Epoch 27/50
16/16 - 0s - loss: 41.1428 - val_loss: 43.8967 - 244m

Epoch 12/50
16/16 - 0s - loss: 47.2377 - val_loss: 50.6269 - 235ms/epoch - 15ms/step
Epoch 13/50
16/16 - 0s - loss: 47.1608 - val_loss: 50.7123 - 224ms/epoch - 14ms/step
Epoch 14/50
16/16 - 0s - loss: 47.0884 - val_loss: 50.7882 - 240ms/epoch - 15ms/step
Epoch 15/50
16/16 - 0s - loss: 47.0414 - val_loss: 50.8545 - 229ms/epoch - 14ms/step
Epoch 16/50
16/16 - 0s - loss: 46.9836 - val_loss: 50.9162 - 252ms/epoch - 16ms/step
Epoch 17/50
16/16 - 0s - loss: 46.9966 - val_loss: 51.0142 - 240ms/epoch - 15ms/step
Epoch 18/50
16/16 - 0s - loss: 46.9234 - val_loss: 51.0544 - 236ms/epoch - 15ms/step
Epoch 19/50
16/16 - 0s - loss: 46.9706 - val_loss: 51.1079 - 232ms/epoch - 14ms/step
Epoch 20/50
16/16 - 0s - loss: 46.9150 - val_loss: 51.1766 - 215ms/epoch - 13ms/step
Epoch 21/50
16/16 - 0s - loss: 46.9015 - val_loss: 51.1996 - 241ms/epoch - 15ms/step
Epoch 22/50
16/16 - 0s - loss: 46.7931 - val_loss: 51.2367 - 244ms/epoch - 15ms/step
Epoch 23/50
16/16 - 0s - loss: 46.7690 - val_loss: 51.3075 - 228m

Epoch 8/50
16/16 - 0s - loss: 45.8809 - val_loss: 47.9181 - 240ms/epoch - 15ms/step
Epoch 9/50
16/16 - 0s - loss: 45.6523 - val_loss: 47.9736 - 235ms/epoch - 15ms/step
Epoch 10/50
16/16 - 0s - loss: 45.6400 - val_loss: 47.9525 - 237ms/epoch - 15ms/step
Epoch 11/50
16/16 - 0s - loss: 45.6102 - val_loss: 47.9623 - 245ms/epoch - 15ms/step
Epoch 12/50
16/16 - 0s - loss: 45.6617 - val_loss: 47.9889 - 234ms/epoch - 15ms/step
Epoch 13/50
16/16 - 0s - loss: 45.5912 - val_loss: 47.9886 - 233ms/epoch - 15ms/step
Epoch 14/50
16/16 - 0s - loss: 45.5689 - val_loss: 47.9786 - 225ms/epoch - 14ms/step
Epoch 15/50
16/16 - 0s - loss: 45.6335 - val_loss: 48.0159 - 239ms/epoch - 15ms/step
Epoch 16/50
16/16 - 0s - loss: 45.5892 - val_loss: 47.9765 - 241ms/epoch - 15ms/step
Epoch 17/50
16/16 - 0s - loss: 45.6381 - val_loss: 48.0125 - 241ms/epoch - 15ms/step
Epoch 18/50
16/16 - 0s - loss: 45.5253 - val_loss: 48.0083 - 232ms/epoch - 15ms/step
Epoch 19/50
16/16 - 0s - loss: 45.5337 - val_loss: 47.9882 - 231ms/

Epoch 4/50
16/16 - 0s - loss: 46.0067 - val_loss: 43.7176 - 214ms/epoch - 13ms/step
Epoch 5/50
16/16 - 0s - loss: 46.0060 - val_loss: 43.7391 - 227ms/epoch - 14ms/step
Epoch 6/50
16/16 - 0s - loss: 45.9828 - val_loss: 43.7301 - 212ms/epoch - 13ms/step
Epoch 7/50
16/16 - 0s - loss: 45.9400 - val_loss: 43.7575 - 209ms/epoch - 13ms/step
Epoch 8/50
16/16 - 0s - loss: 45.9702 - val_loss: 43.7088 - 231ms/epoch - 14ms/step
Epoch 9/50
16/16 - 0s - loss: 45.9256 - val_loss: 43.7230 - 221ms/epoch - 14ms/step
Epoch 10/50
16/16 - 0s - loss: 45.9516 - val_loss: 43.7496 - 233ms/epoch - 15ms/step
Epoch 11/50
16/16 - 0s - loss: 45.8845 - val_loss: 43.6993 - 229ms/epoch - 14ms/step
Epoch 12/50
16/16 - 0s - loss: 45.8980 - val_loss: 43.6870 - 230ms/epoch - 14ms/step
Epoch 13/50
16/16 - 0s - loss: 45.8875 - val_loss: 43.6435 - 208ms/epoch - 13ms/step
Epoch 14/50
16/16 - 0s - loss: 45.9179 - val_loss: 43.6831 - 224ms/epoch - 14ms/step
Epoch 15/50
16/16 - 0s - loss: 45.9081 - val_loss: 43.6504 - 240ms/epoc

Epoch 50/50
16/16 - 0s - loss: 43.2226 - val_loss: 48.4411 - 234ms/epoch - 15ms/step
10/10 [==============================] - 0s 5ms/step
Epoch 1/50
16/16 - 0s - loss: 47.6819 - val_loss: 41.8272 - 316ms/epoch - 20ms/step
Epoch 2/50
16/16 - 0s - loss: 47.5672 - val_loss: 41.7383 - 242ms/epoch - 15ms/step
Epoch 3/50
16/16 - 0s - loss: 47.4807 - val_loss: 41.6852 - 240ms/epoch - 15ms/step
Epoch 4/50
16/16 - 0s - loss: 47.4093 - val_loss: 41.7144 - 224ms/epoch - 14ms/step
Epoch 5/50
16/16 - 0s - loss: 47.3417 - val_loss: 41.6022 - 236ms/epoch - 15ms/step
Epoch 6/50
16/16 - 0s - loss: 47.1165 - val_loss: 41.5676 - 221ms/epoch - 14ms/step
Epoch 7/50
16/16 - 0s - loss: 47.1079 - val_loss: 41.5592 - 232ms/epoch - 14ms/step
Epoch 8/50
16/16 - 0s - loss: 47.1071 - val_loss: 41.5254 - 215ms/epoch - 13ms/step
Epoch 9/50
16/16 - 0s - loss: 46.9615 - val_loss: 41.4718 - 231ms/epoch - 14ms/step
Epoch 10/50
16/16 - 0s - loss: 46.8676 - val_loss: 41.4456 - 236ms/epoch - 15ms/step
Epoch 11/50
16/16 - 0

Epoch 46/50
16/16 - 0s - loss: 40.7183 - val_loss: 50.6995 - 237ms/epoch - 15ms/step
Epoch 47/50
16/16 - 0s - loss: 40.7539 - val_loss: 50.6753 - 233ms/epoch - 15ms/step
Epoch 48/50
16/16 - 0s - loss: 40.6944 - val_loss: 50.7403 - 244ms/epoch - 15ms/step
Epoch 49/50
16/16 - 0s - loss: 40.7951 - val_loss: 50.7087 - 227ms/epoch - 14ms/step
Epoch 50/50
16/16 - 0s - loss: 40.7497 - val_loss: 50.8122 - 240ms/epoch - 15ms/step
10/10 [==============================] - 0s 5ms/step
Epoch 1/50
16/16 - 0s - loss: 42.3462 - val_loss: 52.2292 - 335ms/epoch - 21ms/step
Epoch 2/50
16/16 - 0s - loss: 41.6351 - val_loss: 50.5527 - 231ms/epoch - 14ms/step
Epoch 3/50
16/16 - 0s - loss: 41.3155 - val_loss: 51.0269 - 228ms/epoch - 14ms/step
Epoch 4/50
16/16 - 0s - loss: 41.1325 - val_loss: 51.3116 - 228ms/epoch - 14ms/step
Epoch 5/50
16/16 - 0s - loss: 41.0877 - val_loss: 51.2284 - 222ms/epoch - 14ms/step
Epoch 6/50
16/16 - 0s - loss: 41.0184 - val_loss: 51.2754 - 244ms/epoch - 15ms/step
Epoch 7/50
16/16 -

Epoch 42/50
16/16 - 0s - loss: 42.9126 - val_loss: 42.6111 - 221ms/epoch - 14ms/step
Epoch 43/50
16/16 - 0s - loss: 43.0114 - val_loss: 42.6960 - 244ms/epoch - 15ms/step
Epoch 44/50
16/16 - 0s - loss: 42.8602 - val_loss: 42.6226 - 259ms/epoch - 16ms/step
Epoch 45/50
16/16 - 0s - loss: 42.9221 - val_loss: 42.6439 - 238ms/epoch - 15ms/step
Epoch 46/50
16/16 - 0s - loss: 42.8819 - val_loss: 42.7173 - 239ms/epoch - 15ms/step
Epoch 47/50
16/16 - 0s - loss: 42.9475 - val_loss: 42.7453 - 234ms/epoch - 15ms/step
Epoch 48/50
16/16 - 0s - loss: 42.8742 - val_loss: 42.7053 - 241ms/epoch - 15ms/step
Epoch 49/50
16/16 - 0s - loss: 42.8247 - val_loss: 42.7325 - 226ms/epoch - 14ms/step
Epoch 50/50
16/16 - 0s - loss: 42.8864 - val_loss: 42.8493 - 240ms/epoch - 15ms/step
10/10 [==============================] - 0s 5ms/step
Epoch 1/50
16/16 - 0s - loss: 42.4271 - val_loss: 42.5478 - 341ms/epoch - 21ms/step
Epoch 2/50
16/16 - 0s - loss: 42.0011 - val_loss: 42.5482 - 247ms/epoch - 15ms/step
Epoch 3/50
16/

Epoch 38/50
16/16 - 0s - loss: 43.3336 - val_loss: 47.3249 - 247ms/epoch - 15ms/step
Epoch 39/50
16/16 - 0s - loss: 43.2975 - val_loss: 47.2128 - 234ms/epoch - 15ms/step
Epoch 40/50
16/16 - 0s - loss: 43.2834 - val_loss: 47.2909 - 215ms/epoch - 13ms/step
Epoch 41/50
16/16 - 0s - loss: 43.3158 - val_loss: 47.1301 - 251ms/epoch - 16ms/step
Epoch 42/50
16/16 - 0s - loss: 43.3585 - val_loss: 47.3529 - 225ms/epoch - 14ms/step
Epoch 43/50
16/16 - 0s - loss: 43.4080 - val_loss: 46.9939 - 209ms/epoch - 13ms/step
Epoch 44/50
16/16 - 0s - loss: 43.3684 - val_loss: 47.5751 - 229ms/epoch - 14ms/step
Epoch 45/50
16/16 - 0s - loss: 43.2871 - val_loss: 47.2285 - 202ms/epoch - 13ms/step
Epoch 46/50
16/16 - 0s - loss: 43.2777 - val_loss: 47.2942 - 212ms/epoch - 13ms/step
Epoch 47/50
16/16 - 0s - loss: 43.2302 - val_loss: 47.2124 - 227ms/epoch - 14ms/step
Epoch 48/50
16/16 - 0s - loss: 43.4010 - val_loss: 47.5733 - 230ms/epoch - 14ms/step
Epoch 49/50
16/16 - 0s - loss: 43.2273 - val_loss: 47.1193 - 244m

In [26]:
mse_arr_minmax = np.array(mse_50_Minmax)
mean_mse_minmax = np.mean(mse_arr_minmax)
std_mse_minmax = np.std(mse_arr_minmax)

In [27]:
print('The mean MSE after training the model 50 times after the data is scaled using minmax scaler is: ',mean_mse_minmax,
      ' and the standard deviation is: ', std_mse_minmax)

The mean MSE after training the model 50 times after the data is scaled using minmax scaler is:  74.74712459720027  and the standard deviation is:  45.93906271045597


This performance is better than the standard scaling that was performed before. part A the mean MSE and standard deviation was 58.269382589504986 and 18.803301390418138 respectively which is still better than both the standard scaler and minmax scaler. 

## using the Robust  scaler to scale the data instead of subtracting the mean and dividing by the standard deviation

In [28]:
# Splitting the raw data into training and testing and using the Robust scaler instance to fit and transform the training data
#and using the same transform on the test data:
#Scale features using statistics that are robust to outliers.
#This Scaler removes the median and scales the data according to
#the quantile range (defaults to IQR: Interquartile Range).
#The IQR is the range between the 1st quartile (25th quantile)
#and the 3rd quartile (75th quantile).

mse_50_robust = []
robust = RobustScaler()
model = regression_model()
for i in range(50):
    X_train, X_test, y_train, y_test = train_test_split(predictors, target, test_size=0.30)
    X_train_scaled = robust.fit_transform(X_train)
    X_test_scaled = robust.transform(X_test)
    model.fit(X_train_scaled, y_train, validation_split=0.3, epochs=50, verbose=2)
    predictions = model.predict(X_test_scaled)
    mse_50_robust.append(mean_squared_error(y_test, predictions))






Epoch 1/50
16/16 - 1s - loss: 1563.8579 - val_loss: 1566.5442 - 1s/epoch - 77ms/step
Epoch 2/50
16/16 - 0s - loss: 1549.2405 - val_loss: 1551.7395 - 270ms/epoch - 17ms/step
Epoch 3/50
16/16 - 0s - loss: 1535.1261 - val_loss: 1537.0065 - 287ms/epoch - 18ms/step
Epoch 4/50
16/16 - 0s - loss: 1520.9261 - val_loss: 1522.7347 - 238ms/epoch - 15ms/step
Epoch 5/50
16/16 - 0s - loss: 1506.9801 - val_loss: 1508.5178 - 244ms/epoch - 15ms/step
Epoch 6/50
16/16 - 0s - loss: 1493.0421 - val_loss: 1494.2106 - 247ms/epoch - 15ms/step
Epoch 7/50
16/16 - 0s - loss: 1479.0846 - val_loss: 1479.8419 - 233ms/epoch - 15ms/step
Epoch 8/50
16/16 - 0s - loss: 1465.0063 - val_loss: 1465.0977 - 241ms/epoch - 15ms/step
Epoch 9/50
16/16 - 0s - loss: 1450.4186 - val_loss: 1450.1176 - 242ms/epoch - 15ms/step
Epoch 10/50
16/16 - 0s - loss: 1435.5558 - val_loss: 1434.6239 - 229ms/epoch - 14ms/step
Epoch 11/50
16/16 - 0s - loss: 1420.2705 - val_loss: 1418.3956 - 221ms/epoch - 14ms/step
Epoch 12/50
16/16 - 0s - loss: 14

16/16 - 0s - loss: 157.7826 - val_loss: 119.0160 - 222ms/epoch - 14ms/step
Epoch 45/50
16/16 - 0s - loss: 155.5658 - val_loss: 118.1761 - 222ms/epoch - 14ms/step
Epoch 46/50
16/16 - 0s - loss: 153.6022 - val_loss: 117.4885 - 240ms/epoch - 15ms/step
Epoch 47/50
16/16 - 0s - loss: 151.6017 - val_loss: 116.7407 - 216ms/epoch - 13ms/step
Epoch 48/50
16/16 - 0s - loss: 149.7254 - val_loss: 116.2010 - 217ms/epoch - 14ms/step
Epoch 49/50
16/16 - 0s - loss: 147.9584 - val_loss: 115.6486 - 240ms/epoch - 15ms/step
Epoch 50/50
16/16 - 0s - loss: 146.2102 - val_loss: 115.1165 - 235ms/epoch - 15ms/step
10/10 [==============================] - 0s 4ms/step
Epoch 1/50
16/16 - 0s - loss: 142.5935 - val_loss: 150.6360 - 323ms/epoch - 20ms/step
Epoch 2/50
16/16 - 0s - loss: 141.1586 - val_loss: 149.6635 - 222ms/epoch - 14ms/step
Epoch 3/50
16/16 - 0s - loss: 139.8915 - val_loss: 148.7092 - 240ms/epoch - 15ms/step
Epoch 4/50
16/16 - 0s - loss: 138.4134 - val_loss: 147.8216 - 239ms/epoch - 15ms/step
Epoch 

Epoch 38/50
16/16 - 0s - loss: 90.0313 - val_loss: 79.8514 - 231ms/epoch - 14ms/step
Epoch 39/50
16/16 - 0s - loss: 89.5744 - val_loss: 79.4732 - 231ms/epoch - 14ms/step
Epoch 40/50
16/16 - 0s - loss: 89.0938 - val_loss: 78.9962 - 230ms/epoch - 14ms/step
Epoch 41/50
16/16 - 0s - loss: 88.6827 - val_loss: 78.4146 - 248ms/epoch - 15ms/step
Epoch 42/50
16/16 - 0s - loss: 88.1454 - val_loss: 77.9187 - 228ms/epoch - 14ms/step
Epoch 43/50
16/16 - 0s - loss: 87.6573 - val_loss: 77.4244 - 240ms/epoch - 15ms/step
Epoch 44/50
16/16 - 0s - loss: 87.1375 - val_loss: 76.8978 - 229ms/epoch - 14ms/step
Epoch 45/50
16/16 - 0s - loss: 86.6797 - val_loss: 76.3101 - 234ms/epoch - 15ms/step
Epoch 46/50
16/16 - 0s - loss: 86.1638 - val_loss: 75.8548 - 231ms/epoch - 14ms/step
Epoch 47/50
16/16 - 0s - loss: 85.6890 - val_loss: 75.2620 - 235ms/epoch - 15ms/step
Epoch 48/50
16/16 - 0s - loss: 85.2094 - val_loss: 74.8142 - 242ms/epoch - 15ms/step
Epoch 49/50
16/16 - 0s - loss: 84.7170 - val_loss: 74.1348 - 238m

Epoch 34/50
16/16 - 0s - loss: 55.1727 - val_loss: 50.1241 - 111ms/epoch - 7ms/step
Epoch 35/50
16/16 - 0s - loss: 54.9907 - val_loss: 50.0344 - 100ms/epoch - 6ms/step
Epoch 36/50
16/16 - 0s - loss: 54.7537 - val_loss: 49.7992 - 104ms/epoch - 6ms/step
Epoch 37/50
16/16 - 0s - loss: 54.5140 - val_loss: 49.5626 - 103ms/epoch - 6ms/step
Epoch 38/50
16/16 - 0s - loss: 54.3759 - val_loss: 49.3584 - 113ms/epoch - 7ms/step
Epoch 39/50
16/16 - 0s - loss: 54.1759 - val_loss: 49.3048 - 112ms/epoch - 7ms/step
Epoch 40/50
16/16 - 0s - loss: 54.0015 - val_loss: 49.1678 - 121ms/epoch - 8ms/step
Epoch 41/50
16/16 - 0s - loss: 53.8641 - val_loss: 49.0638 - 98ms/epoch - 6ms/step
Epoch 42/50
16/16 - 0s - loss: 53.6425 - val_loss: 48.8246 - 100ms/epoch - 6ms/step
Epoch 43/50
16/16 - 0s - loss: 53.4934 - val_loss: 48.8024 - 97ms/epoch - 6ms/step
Epoch 44/50
16/16 - 0s - loss: 53.3514 - val_loss: 48.5494 - 95ms/epoch - 6ms/step
Epoch 45/50
16/16 - 0s - loss: 53.1736 - val_loss: 48.3928 - 99ms/epoch - 6ms/s

Epoch 31/50
16/16 - 0s - loss: 43.5251 - val_loss: 45.1088 - 103ms/epoch - 6ms/step
Epoch 32/50
16/16 - 0s - loss: 43.4394 - val_loss: 45.1393 - 112ms/epoch - 7ms/step
Epoch 33/50
16/16 - 0s - loss: 43.4538 - val_loss: 45.1676 - 104ms/epoch - 6ms/step
Epoch 34/50
16/16 - 0s - loss: 43.3224 - val_loss: 45.0864 - 118ms/epoch - 7ms/step
Epoch 35/50
16/16 - 0s - loss: 43.3149 - val_loss: 45.0599 - 133ms/epoch - 8ms/step
Epoch 36/50
16/16 - 0s - loss: 43.2829 - val_loss: 45.0273 - 117ms/epoch - 7ms/step
Epoch 37/50
16/16 - 0s - loss: 43.2412 - val_loss: 44.9628 - 117ms/epoch - 7ms/step
Epoch 38/50
16/16 - 0s - loss: 43.1890 - val_loss: 44.9611 - 108ms/epoch - 7ms/step
Epoch 39/50
16/16 - 0s - loss: 43.1888 - val_loss: 44.8361 - 104ms/epoch - 6ms/step
Epoch 40/50
16/16 - 0s - loss: 43.1471 - val_loss: 44.8450 - 109ms/epoch - 7ms/step
Epoch 41/50
16/16 - 0s - loss: 43.0986 - val_loss: 44.8010 - 108ms/epoch - 7ms/step
Epoch 42/50
16/16 - 0s - loss: 43.1250 - val_loss: 44.9070 - 111ms/epoch - 7

Epoch 28/50
16/16 - 0s - loss: 41.2079 - val_loss: 48.2576 - 125ms/epoch - 8ms/step
Epoch 29/50
16/16 - 0s - loss: 41.1845 - val_loss: 48.1782 - 113ms/epoch - 7ms/step
Epoch 30/50
16/16 - 0s - loss: 41.1335 - val_loss: 48.1269 - 118ms/epoch - 7ms/step
Epoch 31/50
16/16 - 0s - loss: 41.1110 - val_loss: 48.2743 - 132ms/epoch - 8ms/step
Epoch 32/50
16/16 - 0s - loss: 41.1060 - val_loss: 48.1741 - 118ms/epoch - 7ms/step
Epoch 33/50
16/16 - 0s - loss: 41.1254 - val_loss: 48.1976 - 114ms/epoch - 7ms/step
Epoch 34/50
16/16 - 0s - loss: 41.0218 - val_loss: 47.9049 - 109ms/epoch - 7ms/step
Epoch 35/50
16/16 - 0s - loss: 40.9573 - val_loss: 48.1069 - 108ms/epoch - 7ms/step
Epoch 36/50
16/16 - 0s - loss: 40.9136 - val_loss: 48.0999 - 108ms/epoch - 7ms/step
Epoch 37/50
16/16 - 0s - loss: 40.8823 - val_loss: 48.1846 - 107ms/epoch - 7ms/step
Epoch 38/50
16/16 - 0s - loss: 40.8542 - val_loss: 48.0855 - 107ms/epoch - 7ms/step
Epoch 39/50
16/16 - 0s - loss: 40.8203 - val_loss: 48.0890 - 112ms/epoch - 7

Epoch 25/50
16/16 - 0s - loss: 37.7319 - val_loss: 37.2305 - 98ms/epoch - 6ms/step
Epoch 26/50
16/16 - 0s - loss: 37.7742 - val_loss: 37.2481 - 101ms/epoch - 6ms/step
Epoch 27/50
16/16 - 0s - loss: 37.6194 - val_loss: 37.3379 - 113ms/epoch - 7ms/step
Epoch 28/50
16/16 - 0s - loss: 37.5700 - val_loss: 37.3719 - 102ms/epoch - 6ms/step
Epoch 29/50
16/16 - 0s - loss: 37.5917 - val_loss: 37.3796 - 122ms/epoch - 8ms/step
Epoch 30/50
16/16 - 0s - loss: 37.4920 - val_loss: 37.4008 - 112ms/epoch - 7ms/step
Epoch 31/50
16/16 - 0s - loss: 37.5369 - val_loss: 37.4912 - 110ms/epoch - 7ms/step
Epoch 32/50
16/16 - 0s - loss: 37.4460 - val_loss: 37.4391 - 107ms/epoch - 7ms/step
Epoch 33/50
16/16 - 0s - loss: 37.4348 - val_loss: 37.4817 - 107ms/epoch - 7ms/step
Epoch 34/50
16/16 - 0s - loss: 37.4070 - val_loss: 37.5764 - 103ms/epoch - 6ms/step
Epoch 35/50
16/16 - 0s - loss: 37.3633 - val_loss: 37.5872 - 113ms/epoch - 7ms/step
Epoch 36/50
16/16 - 0s - loss: 37.3328 - val_loss: 37.5963 - 112ms/epoch - 7m

Epoch 22/50
16/16 - 0s - loss: 35.0791 - val_loss: 42.0604 - 107ms/epoch - 7ms/step
Epoch 23/50
16/16 - 0s - loss: 35.0289 - val_loss: 42.2077 - 99ms/epoch - 6ms/step
Epoch 24/50
16/16 - 0s - loss: 35.0074 - val_loss: 42.0472 - 117ms/epoch - 7ms/step
Epoch 25/50
16/16 - 0s - loss: 34.9239 - val_loss: 42.0805 - 101ms/epoch - 6ms/step
Epoch 26/50
16/16 - 0s - loss: 34.9185 - val_loss: 41.8357 - 107ms/epoch - 7ms/step
Epoch 27/50
16/16 - 0s - loss: 34.8680 - val_loss: 41.9696 - 104ms/epoch - 7ms/step
Epoch 28/50
16/16 - 0s - loss: 34.8289 - val_loss: 42.0464 - 111ms/epoch - 7ms/step
Epoch 29/50
16/16 - 0s - loss: 34.8018 - val_loss: 41.9251 - 111ms/epoch - 7ms/step
Epoch 30/50
16/16 - 0s - loss: 34.7658 - val_loss: 41.8648 - 111ms/epoch - 7ms/step
Epoch 31/50
16/16 - 0s - loss: 34.7265 - val_loss: 41.7896 - 101ms/epoch - 6ms/step
Epoch 32/50
16/16 - 0s - loss: 34.6882 - val_loss: 41.8112 - 104ms/epoch - 7ms/step
Epoch 33/50
16/16 - 0s - loss: 34.7604 - val_loss: 42.1954 - 109ms/epoch - 7m

Epoch 19/50
16/16 - 0s - loss: 35.5615 - val_loss: 47.3952 - 103ms/epoch - 6ms/step
Epoch 20/50
16/16 - 0s - loss: 35.5365 - val_loss: 47.3540 - 101ms/epoch - 6ms/step
Epoch 21/50
16/16 - 0s - loss: 35.4785 - val_loss: 47.4858 - 102ms/epoch - 6ms/step
Epoch 22/50
16/16 - 0s - loss: 35.4178 - val_loss: 47.3904 - 115ms/epoch - 7ms/step
Epoch 23/50
16/16 - 0s - loss: 35.4081 - val_loss: 47.2751 - 115ms/epoch - 7ms/step
Epoch 24/50
16/16 - 0s - loss: 35.3717 - val_loss: 47.4527 - 108ms/epoch - 7ms/step
Epoch 25/50
16/16 - 0s - loss: 35.3126 - val_loss: 47.4328 - 112ms/epoch - 7ms/step
Epoch 26/50
16/16 - 0s - loss: 35.2784 - val_loss: 47.4597 - 114ms/epoch - 7ms/step
Epoch 27/50
16/16 - 0s - loss: 35.2286 - val_loss: 47.3636 - 102ms/epoch - 6ms/step
Epoch 28/50
16/16 - 0s - loss: 35.2084 - val_loss: 47.5405 - 106ms/epoch - 7ms/step
Epoch 29/50
16/16 - 0s - loss: 35.1728 - val_loss: 47.4822 - 101ms/epoch - 6ms/step
Epoch 30/50
16/16 - 0s - loss: 35.1366 - val_loss: 47.5399 - 97ms/epoch - 6m

Epoch 16/50
16/16 - 0s - loss: 37.2872 - val_loss: 41.2576 - 130ms/epoch - 8ms/step
Epoch 17/50
16/16 - 0s - loss: 37.2159 - val_loss: 41.2976 - 122ms/epoch - 8ms/step
Epoch 18/50
16/16 - 0s - loss: 37.2243 - val_loss: 41.2492 - 118ms/epoch - 7ms/step
Epoch 19/50
16/16 - 0s - loss: 37.1399 - val_loss: 41.2158 - 115ms/epoch - 7ms/step
Epoch 20/50
16/16 - 0s - loss: 37.0918 - val_loss: 41.2865 - 112ms/epoch - 7ms/step
Epoch 21/50
16/16 - 0s - loss: 37.0541 - val_loss: 41.3311 - 113ms/epoch - 7ms/step
Epoch 22/50
16/16 - 0s - loss: 37.0349 - val_loss: 41.3122 - 107ms/epoch - 7ms/step
Epoch 23/50
16/16 - 0s - loss: 37.0843 - val_loss: 41.3921 - 106ms/epoch - 7ms/step
Epoch 24/50
16/16 - 0s - loss: 36.9974 - val_loss: 41.3941 - 116ms/epoch - 7ms/step
Epoch 25/50
16/16 - 0s - loss: 36.9445 - val_loss: 41.3994 - 112ms/epoch - 7ms/step
Epoch 26/50
16/16 - 0s - loss: 36.9290 - val_loss: 41.2719 - 106ms/epoch - 7ms/step
Epoch 27/50
16/16 - 0s - loss: 36.9018 - val_loss: 41.2391 - 110ms/epoch - 7

Epoch 13/50
16/16 - 0s - loss: 35.6090 - val_loss: 36.8320 - 106ms/epoch - 7ms/step
Epoch 14/50
16/16 - 0s - loss: 35.5842 - val_loss: 36.8451 - 109ms/epoch - 7ms/step
Epoch 15/50
16/16 - 0s - loss: 35.5483 - val_loss: 36.9739 - 118ms/epoch - 7ms/step
Epoch 16/50
16/16 - 0s - loss: 35.5394 - val_loss: 36.9744 - 99ms/epoch - 6ms/step
Epoch 17/50
16/16 - 0s - loss: 35.4808 - val_loss: 36.9813 - 112ms/epoch - 7ms/step
Epoch 18/50
16/16 - 0s - loss: 35.4739 - val_loss: 37.0734 - 120ms/epoch - 8ms/step
Epoch 19/50
16/16 - 0s - loss: 35.4521 - val_loss: 37.1166 - 104ms/epoch - 6ms/step
Epoch 20/50
16/16 - 0s - loss: 35.4109 - val_loss: 36.9330 - 99ms/epoch - 6ms/step
Epoch 21/50
16/16 - 0s - loss: 35.3965 - val_loss: 37.0609 - 104ms/epoch - 6ms/step
Epoch 22/50
16/16 - 0s - loss: 35.3818 - val_loss: 37.2143 - 105ms/epoch - 7ms/step
Epoch 23/50
16/16 - 0s - loss: 35.3839 - val_loss: 37.2266 - 109ms/epoch - 7ms/step
Epoch 24/50
16/16 - 0s - loss: 35.3307 - val_loss: 36.9050 - 112ms/epoch - 7ms

Epoch 10/50
16/16 - 0s - loss: 37.0579 - val_loss: 33.7853 - 113ms/epoch - 7ms/step
Epoch 11/50
16/16 - 0s - loss: 37.0140 - val_loss: 33.7940 - 121ms/epoch - 8ms/step
Epoch 12/50
16/16 - 0s - loss: 36.9957 - val_loss: 33.8552 - 113ms/epoch - 7ms/step
Epoch 13/50
16/16 - 0s - loss: 36.9427 - val_loss: 33.8498 - 111ms/epoch - 7ms/step
Epoch 14/50
16/16 - 0s - loss: 36.9082 - val_loss: 33.8785 - 112ms/epoch - 7ms/step
Epoch 15/50
16/16 - 0s - loss: 36.8646 - val_loss: 33.8911 - 118ms/epoch - 7ms/step
Epoch 16/50
16/16 - 0s - loss: 36.8280 - val_loss: 33.8190 - 110ms/epoch - 7ms/step
Epoch 17/50
16/16 - 0s - loss: 36.7846 - val_loss: 33.8645 - 128ms/epoch - 8ms/step
Epoch 18/50
16/16 - 0s - loss: 36.7193 - val_loss: 33.8842 - 121ms/epoch - 8ms/step
Epoch 19/50
16/16 - 0s - loss: 36.7110 - val_loss: 33.8979 - 88ms/epoch - 5ms/step
Epoch 20/50
16/16 - 0s - loss: 36.6474 - val_loss: 33.8869 - 123ms/epoch - 8ms/step
Epoch 21/50
16/16 - 0s - loss: 36.6235 - val_loss: 33.8500 - 110ms/epoch - 7m

Epoch 7/50
16/16 - 0s - loss: 35.4269 - val_loss: 38.9376 - 113ms/epoch - 7ms/step
Epoch 8/50
16/16 - 0s - loss: 35.3607 - val_loss: 38.8732 - 113ms/epoch - 7ms/step
Epoch 9/50
16/16 - 0s - loss: 35.2669 - val_loss: 39.0100 - 102ms/epoch - 6ms/step
Epoch 10/50
16/16 - 0s - loss: 35.2224 - val_loss: 38.9608 - 122ms/epoch - 8ms/step
Epoch 11/50
16/16 - 0s - loss: 35.1237 - val_loss: 39.1563 - 102ms/epoch - 6ms/step
Epoch 12/50
16/16 - 0s - loss: 35.0845 - val_loss: 39.2842 - 108ms/epoch - 7ms/step
Epoch 13/50
16/16 - 0s - loss: 35.0037 - val_loss: 39.3352 - 110ms/epoch - 7ms/step
Epoch 14/50
16/16 - 0s - loss: 34.9914 - val_loss: 39.4065 - 105ms/epoch - 7ms/step
Epoch 15/50
16/16 - 0s - loss: 34.9322 - val_loss: 39.3903 - 120ms/epoch - 7ms/step
Epoch 16/50
16/16 - 0s - loss: 34.8771 - val_loss: 39.3892 - 121ms/epoch - 8ms/step
Epoch 17/50
16/16 - 0s - loss: 34.8643 - val_loss: 39.2943 - 107ms/epoch - 7ms/step
Epoch 18/50
16/16 - 0s - loss: 34.8082 - val_loss: 39.3980 - 114ms/epoch - 7ms/

Epoch 4/50
16/16 - 0s - loss: 34.4142 - val_loss: 32.7185 - 107ms/epoch - 7ms/step
Epoch 5/50
16/16 - 0s - loss: 34.2459 - val_loss: 32.6194 - 109ms/epoch - 7ms/step
Epoch 6/50
16/16 - 0s - loss: 34.0268 - val_loss: 32.8773 - 115ms/epoch - 7ms/step
Epoch 7/50
16/16 - 0s - loss: 33.9302 - val_loss: 33.1056 - 116ms/epoch - 7ms/step
Epoch 8/50
16/16 - 0s - loss: 33.8391 - val_loss: 33.0200 - 103ms/epoch - 6ms/step
Epoch 9/50
16/16 - 0s - loss: 33.7608 - val_loss: 33.0949 - 107ms/epoch - 7ms/step
Epoch 10/50
16/16 - 0s - loss: 33.6912 - val_loss: 33.2847 - 124ms/epoch - 8ms/step
Epoch 11/50
16/16 - 0s - loss: 33.6750 - val_loss: 33.3747 - 99ms/epoch - 6ms/step
Epoch 12/50
16/16 - 0s - loss: 33.6295 - val_loss: 33.2426 - 108ms/epoch - 7ms/step
Epoch 13/50
16/16 - 0s - loss: 33.5642 - val_loss: 33.2606 - 102ms/epoch - 6ms/step
Epoch 14/50
16/16 - 0s - loss: 33.5572 - val_loss: 33.4662 - 104ms/epoch - 6ms/step
Epoch 15/50
16/16 - 0s - loss: 33.5097 - val_loss: 33.5545 - 121ms/epoch - 8ms/step

10/10 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 36.6227 - val_loss: 34.3545 - 150ms/epoch - 9ms/step
Epoch 2/50
16/16 - 0s - loss: 35.4222 - val_loss: 33.2583 - 113ms/epoch - 7ms/step
Epoch 3/50
16/16 - 0s - loss: 34.6507 - val_loss: 32.6179 - 114ms/epoch - 7ms/step
Epoch 4/50
16/16 - 0s - loss: 34.2321 - val_loss: 32.1976 - 112ms/epoch - 7ms/step
Epoch 5/50
16/16 - 0s - loss: 34.0159 - val_loss: 31.9871 - 109ms/epoch - 7ms/step
Epoch 6/50
16/16 - 0s - loss: 33.8297 - val_loss: 31.8310 - 103ms/epoch - 6ms/step
Epoch 7/50
16/16 - 0s - loss: 33.7605 - val_loss: 31.8513 - 111ms/epoch - 7ms/step
Epoch 8/50
16/16 - 0s - loss: 33.5136 - val_loss: 31.7875 - 101ms/epoch - 6ms/step
Epoch 9/50
16/16 - 0s - loss: 33.4419 - val_loss: 31.6312 - 115ms/epoch - 7ms/step
Epoch 10/50
16/16 - 0s - loss: 33.3207 - val_loss: 31.6560 - 134ms/epoch - 8ms/step
Epoch 11/50
16/16 - 0s - loss: 33.2416 - val_loss: 31.5572 - 114ms/epoch - 7ms/step
Epoch 12/50
16/16 - 0s - loss: 3

Epoch 48/50
16/16 - 0s - loss: 35.2042 - val_loss: 37.0792 - 137ms/epoch - 9ms/step
Epoch 49/50
16/16 - 0s - loss: 35.1898 - val_loss: 37.1449 - 123ms/epoch - 8ms/step
Epoch 50/50
16/16 - 0s - loss: 35.1800 - val_loss: 37.0920 - 117ms/epoch - 7ms/step
10/10 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 44.1492 - val_loss: 39.8205 - 161ms/epoch - 10ms/step
Epoch 2/50
16/16 - 0s - loss: 34.1273 - val_loss: 36.6112 - 119ms/epoch - 7ms/step
Epoch 3/50
16/16 - 0s - loss: 31.0705 - val_loss: 37.3284 - 115ms/epoch - 7ms/step
Epoch 4/50
16/16 - 0s - loss: 30.6948 - val_loss: 37.6091 - 111ms/epoch - 7ms/step
Epoch 5/50
16/16 - 0s - loss: 30.5806 - val_loss: 37.3633 - 119ms/epoch - 7ms/step
Epoch 6/50
16/16 - 0s - loss: 30.2851 - val_loss: 37.6690 - 108ms/epoch - 7ms/step
Epoch 7/50
16/16 - 0s - loss: 30.2589 - val_loss: 37.6758 - 122ms/epoch - 8ms/step
Epoch 8/50
16/16 - 0s - loss: 30.0590 - val_loss: 37.6474 - 117ms/epoch - 7ms/step
Epoch 9/50
16/16 - 0s - loss: 

Epoch 45/50
16/16 - 0s - loss: 36.3169 - val_loss: 36.9279 - 108ms/epoch - 7ms/step
Epoch 46/50
16/16 - 0s - loss: 36.3553 - val_loss: 36.9614 - 113ms/epoch - 7ms/step
Epoch 47/50
16/16 - 0s - loss: 36.3314 - val_loss: 36.8784 - 105ms/epoch - 7ms/step
Epoch 48/50
16/16 - 0s - loss: 36.3038 - val_loss: 36.7870 - 110ms/epoch - 7ms/step
Epoch 49/50
16/16 - 0s - loss: 36.2836 - val_loss: 36.8132 - 112ms/epoch - 7ms/step
Epoch 50/50
16/16 - 0s - loss: 36.2936 - val_loss: 36.8147 - 111ms/epoch - 7ms/step
10/10 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 38.5056 - val_loss: 37.1356 - 142ms/epoch - 9ms/step
Epoch 2/50
16/16 - 0s - loss: 36.3911 - val_loss: 35.6430 - 113ms/epoch - 7ms/step
Epoch 3/50
16/16 - 0s - loss: 35.5633 - val_loss: 35.3326 - 116ms/epoch - 7ms/step
Epoch 4/50
16/16 - 0s - loss: 35.2465 - val_loss: 35.3991 - 107ms/epoch - 7ms/step
Epoch 5/50
16/16 - 0s - loss: 35.0309 - val_loss: 35.6901 - 119ms/epoch - 7ms/step
Epoch 6/50
16/16 - 0s - loss

Epoch 42/50
16/16 - 0s - loss: 35.5390 - val_loss: 32.2138 - 135ms/epoch - 8ms/step
Epoch 43/50
16/16 - 0s - loss: 35.4396 - val_loss: 32.3731 - 115ms/epoch - 7ms/step
Epoch 44/50
16/16 - 0s - loss: 35.4537 - val_loss: 32.2541 - 137ms/epoch - 9ms/step
Epoch 45/50
16/16 - 0s - loss: 35.4413 - val_loss: 32.3286 - 121ms/epoch - 8ms/step
Epoch 46/50
16/16 - 0s - loss: 35.4711 - val_loss: 32.5309 - 108ms/epoch - 7ms/step
Epoch 47/50
16/16 - 0s - loss: 35.4767 - val_loss: 32.8108 - 105ms/epoch - 7ms/step
Epoch 48/50
16/16 - 0s - loss: 35.3798 - val_loss: 32.4455 - 106ms/epoch - 7ms/step
Epoch 49/50
16/16 - 0s - loss: 35.3665 - val_loss: 32.4823 - 110ms/epoch - 7ms/step
Epoch 50/50
16/16 - 0s - loss: 35.4000 - val_loss: 32.6415 - 128ms/epoch - 8ms/step
10/10 [==============================] - 0s 3ms/step
Epoch 1/50
16/16 - 0s - loss: 37.5827 - val_loss: 43.6109 - 174ms/epoch - 11ms/step
Epoch 2/50
16/16 - 0s - loss: 36.6392 - val_loss: 42.2697 - 144ms/epoch - 9ms/step
Epoch 3/50
16/16 - 0s - 

Epoch 39/50
16/16 - 0s - loss: 34.9996 - val_loss: 32.2741 - 98ms/epoch - 6ms/step
Epoch 40/50
16/16 - 0s - loss: 35.0742 - val_loss: 32.2047 - 105ms/epoch - 7ms/step
Epoch 41/50
16/16 - 0s - loss: 34.9857 - val_loss: 32.2804 - 106ms/epoch - 7ms/step
Epoch 42/50
16/16 - 0s - loss: 34.9628 - val_loss: 32.2867 - 101ms/epoch - 6ms/step
Epoch 43/50
16/16 - 0s - loss: 34.9840 - val_loss: 32.2630 - 107ms/epoch - 7ms/step
Epoch 44/50
16/16 - 0s - loss: 34.9469 - val_loss: 32.2421 - 113ms/epoch - 7ms/step
Epoch 45/50
16/16 - 0s - loss: 34.9391 - val_loss: 32.2911 - 110ms/epoch - 7ms/step
Epoch 46/50
16/16 - 0s - loss: 34.9528 - val_loss: 32.3897 - 110ms/epoch - 7ms/step
Epoch 47/50
16/16 - 0s - loss: 34.9269 - val_loss: 32.3360 - 101ms/epoch - 6ms/step
Epoch 48/50
16/16 - 0s - loss: 34.8966 - val_loss: 32.3138 - 99ms/epoch - 6ms/step
Epoch 49/50
16/16 - 0s - loss: 34.9324 - val_loss: 32.3714 - 107ms/epoch - 7ms/step
Epoch 50/50
16/16 - 0s - loss: 34.9097 - val_loss: 32.2432 - 106ms/epoch - 7ms

Epoch 36/50
16/16 - 0s - loss: 34.0414 - val_loss: 32.6392 - 93ms/epoch - 6ms/step
Epoch 37/50
16/16 - 0s - loss: 34.0522 - val_loss: 32.6004 - 123ms/epoch - 8ms/step
Epoch 38/50
16/16 - 0s - loss: 34.0497 - val_loss: 32.5220 - 129ms/epoch - 8ms/step
Epoch 39/50
16/16 - 0s - loss: 34.0414 - val_loss: 32.6726 - 117ms/epoch - 7ms/step
Epoch 40/50
16/16 - 0s - loss: 34.0228 - val_loss: 32.6735 - 103ms/epoch - 6ms/step
Epoch 41/50
16/16 - 0s - loss: 33.9540 - val_loss: 32.6642 - 137ms/epoch - 9ms/step
Epoch 42/50
16/16 - 0s - loss: 33.9607 - val_loss: 32.7686 - 112ms/epoch - 7ms/step
Epoch 43/50
16/16 - 0s - loss: 33.9340 - val_loss: 32.7331 - 128ms/epoch - 8ms/step
Epoch 44/50
16/16 - 0s - loss: 33.9463 - val_loss: 32.6980 - 106ms/epoch - 7ms/step
Epoch 45/50
16/16 - 0s - loss: 33.9350 - val_loss: 32.7414 - 105ms/epoch - 7ms/step
Epoch 46/50
16/16 - 0s - loss: 33.9049 - val_loss: 32.7940 - 111ms/epoch - 7ms/step
Epoch 47/50
16/16 - 0s - loss: 33.9084 - val_loss: 32.7370 - 108ms/epoch - 7m

Epoch 33/50
16/16 - 0s - loss: 32.1313 - val_loss: 38.0969 - 107ms/epoch - 7ms/step
Epoch 34/50
16/16 - 0s - loss: 32.1391 - val_loss: 37.9988 - 101ms/epoch - 6ms/step
Epoch 35/50
16/16 - 0s - loss: 32.0713 - val_loss: 38.0721 - 124ms/epoch - 8ms/step
Epoch 36/50
16/16 - 0s - loss: 32.0739 - val_loss: 38.1004 - 109ms/epoch - 7ms/step
Epoch 37/50
16/16 - 0s - loss: 32.0416 - val_loss: 38.2076 - 108ms/epoch - 7ms/step
Epoch 38/50
16/16 - 0s - loss: 32.0288 - val_loss: 38.1257 - 117ms/epoch - 7ms/step
Epoch 39/50
16/16 - 0s - loss: 32.0454 - val_loss: 38.2173 - 113ms/epoch - 7ms/step
Epoch 40/50
16/16 - 0s - loss: 32.0105 - val_loss: 38.1697 - 108ms/epoch - 7ms/step
Epoch 41/50
16/16 - 0s - loss: 32.0409 - val_loss: 38.1304 - 107ms/epoch - 7ms/step
Epoch 42/50
16/16 - 0s - loss: 32.0057 - val_loss: 38.2794 - 115ms/epoch - 7ms/step
Epoch 43/50
16/16 - 0s - loss: 31.9390 - val_loss: 38.2722 - 113ms/epoch - 7ms/step
Epoch 44/50
16/16 - 0s - loss: 32.0114 - val_loss: 38.1901 - 108ms/epoch - 7

Epoch 30/50
16/16 - 0s - loss: 35.1788 - val_loss: 32.0948 - 114ms/epoch - 7ms/step
Epoch 31/50
16/16 - 0s - loss: 35.1214 - val_loss: 32.1388 - 121ms/epoch - 8ms/step
Epoch 32/50
16/16 - 0s - loss: 35.0994 - val_loss: 32.1169 - 119ms/epoch - 7ms/step
Epoch 33/50
16/16 - 0s - loss: 35.0965 - val_loss: 32.1620 - 110ms/epoch - 7ms/step
Epoch 34/50
16/16 - 0s - loss: 35.0422 - val_loss: 32.1185 - 109ms/epoch - 7ms/step
Epoch 35/50
16/16 - 0s - loss: 35.0159 - val_loss: 32.1073 - 112ms/epoch - 7ms/step
Epoch 36/50
16/16 - 0s - loss: 34.9806 - val_loss: 32.1241 - 108ms/epoch - 7ms/step
Epoch 37/50
16/16 - 0s - loss: 34.9976 - val_loss: 32.1348 - 128ms/epoch - 8ms/step
Epoch 38/50
16/16 - 0s - loss: 34.9167 - val_loss: 32.1552 - 115ms/epoch - 7ms/step
Epoch 39/50
16/16 - 0s - loss: 34.9155 - val_loss: 32.1833 - 157ms/epoch - 10ms/step
Epoch 40/50
16/16 - 0s - loss: 34.9194 - val_loss: 32.1640 - 125ms/epoch - 8ms/step
Epoch 41/50
16/16 - 0s - loss: 34.8598 - val_loss: 32.1549 - 112ms/epoch - 

Epoch 27/50
16/16 - 0s - loss: 33.6756 - val_loss: 33.7637 - 114ms/epoch - 7ms/step
Epoch 28/50
16/16 - 0s - loss: 33.5483 - val_loss: 33.6567 - 132ms/epoch - 8ms/step
Epoch 29/50
16/16 - 0s - loss: 33.5369 - val_loss: 33.6108 - 134ms/epoch - 8ms/step
Epoch 30/50
16/16 - 0s - loss: 33.5623 - val_loss: 33.7391 - 123ms/epoch - 8ms/step
Epoch 31/50
16/16 - 0s - loss: 33.5871 - val_loss: 33.7073 - 109ms/epoch - 7ms/step
Epoch 32/50
16/16 - 0s - loss: 33.6233 - val_loss: 33.6871 - 125ms/epoch - 8ms/step
Epoch 33/50
16/16 - 0s - loss: 33.4978 - val_loss: 33.7819 - 105ms/epoch - 7ms/step
Epoch 34/50
16/16 - 0s - loss: 33.5578 - val_loss: 33.9578 - 114ms/epoch - 7ms/step
Epoch 35/50
16/16 - 0s - loss: 33.4728 - val_loss: 33.7889 - 125ms/epoch - 8ms/step
Epoch 36/50
16/16 - 0s - loss: 33.4674 - val_loss: 33.8364 - 105ms/epoch - 7ms/step
Epoch 37/50
16/16 - 0s - loss: 33.5755 - val_loss: 33.9163 - 131ms/epoch - 8ms/step
Epoch 38/50
16/16 - 0s - loss: 33.5003 - val_loss: 34.0549 - 116ms/epoch - 7

Epoch 24/50
16/16 - 0s - loss: 34.3242 - val_loss: 35.4187 - 116ms/epoch - 7ms/step
Epoch 25/50
16/16 - 0s - loss: 34.3378 - val_loss: 35.3167 - 111ms/epoch - 7ms/step
Epoch 26/50
16/16 - 0s - loss: 34.2914 - val_loss: 35.2374 - 128ms/epoch - 8ms/step
Epoch 27/50
16/16 - 0s - loss: 34.2778 - val_loss: 35.2304 - 115ms/epoch - 7ms/step
Epoch 28/50
16/16 - 0s - loss: 34.2731 - val_loss: 35.3494 - 116ms/epoch - 7ms/step
Epoch 29/50
16/16 - 0s - loss: 34.2302 - val_loss: 35.2972 - 120ms/epoch - 8ms/step
Epoch 30/50
16/16 - 0s - loss: 34.1572 - val_loss: 35.2752 - 120ms/epoch - 8ms/step
Epoch 31/50
16/16 - 0s - loss: 34.1253 - val_loss: 35.2448 - 111ms/epoch - 7ms/step
Epoch 32/50
16/16 - 0s - loss: 34.1256 - val_loss: 35.1546 - 104ms/epoch - 6ms/step
Epoch 33/50
16/16 - 0s - loss: 34.0838 - val_loss: 35.2571 - 111ms/epoch - 7ms/step
Epoch 34/50
16/16 - 0s - loss: 34.0927 - val_loss: 35.1788 - 112ms/epoch - 7ms/step
Epoch 35/50
16/16 - 0s - loss: 34.0527 - val_loss: 35.1860 - 112ms/epoch - 7

Epoch 21/50
16/16 - 0s - loss: 32.6492 - val_loss: 29.7204 - 110ms/epoch - 7ms/step
Epoch 22/50
16/16 - 0s - loss: 32.6184 - val_loss: 29.7931 - 106ms/epoch - 7ms/step
Epoch 23/50
16/16 - 0s - loss: 32.5748 - val_loss: 29.7205 - 108ms/epoch - 7ms/step
Epoch 24/50
16/16 - 0s - loss: 32.4990 - val_loss: 29.7398 - 135ms/epoch - 8ms/step
Epoch 25/50
16/16 - 0s - loss: 32.4794 - val_loss: 29.8039 - 133ms/epoch - 8ms/step
Epoch 26/50
16/16 - 0s - loss: 32.4703 - val_loss: 29.8388 - 101ms/epoch - 6ms/step
Epoch 27/50
16/16 - 0s - loss: 32.4916 - val_loss: 29.8073 - 109ms/epoch - 7ms/step
Epoch 28/50
16/16 - 0s - loss: 32.4645 - val_loss: 29.7839 - 113ms/epoch - 7ms/step
Epoch 29/50
16/16 - 0s - loss: 32.3689 - val_loss: 29.8706 - 103ms/epoch - 6ms/step
Epoch 30/50
16/16 - 0s - loss: 32.3097 - val_loss: 29.9028 - 120ms/epoch - 7ms/step
Epoch 31/50
16/16 - 0s - loss: 32.2775 - val_loss: 29.8874 - 114ms/epoch - 7ms/step
Epoch 32/50
16/16 - 0s - loss: 32.2688 - val_loss: 29.8564 - 112ms/epoch - 7

Epoch 18/50
16/16 - 0s - loss: 30.9628 - val_loss: 34.0794 - 109ms/epoch - 7ms/step
Epoch 19/50
16/16 - 0s - loss: 30.9501 - val_loss: 34.0219 - 110ms/epoch - 7ms/step
Epoch 20/50
16/16 - 0s - loss: 30.8754 - val_loss: 34.2055 - 128ms/epoch - 8ms/step
Epoch 21/50
16/16 - 0s - loss: 30.8115 - val_loss: 34.2944 - 108ms/epoch - 7ms/step
Epoch 22/50
16/16 - 0s - loss: 30.7960 - val_loss: 34.3002 - 119ms/epoch - 7ms/step
Epoch 23/50
16/16 - 0s - loss: 30.7785 - val_loss: 34.3282 - 110ms/epoch - 7ms/step
Epoch 24/50
16/16 - 0s - loss: 30.7401 - val_loss: 34.2820 - 108ms/epoch - 7ms/step
Epoch 25/50
16/16 - 0s - loss: 30.6283 - val_loss: 34.3630 - 106ms/epoch - 7ms/step
Epoch 26/50
16/16 - 0s - loss: 30.5886 - val_loss: 34.3366 - 106ms/epoch - 7ms/step
Epoch 27/50
16/16 - 0s - loss: 30.6355 - val_loss: 34.2309 - 110ms/epoch - 7ms/step
Epoch 28/50
16/16 - 0s - loss: 30.5062 - val_loss: 34.3768 - 103ms/epoch - 6ms/step
Epoch 29/50
16/16 - 0s - loss: 30.4972 - val_loss: 34.4116 - 127ms/epoch - 8

In [33]:
mse_arr_robust = np.array(mse_50_robust)
mean_mse_robust = np.mean(mse_arr_robust)
std_mse_robust = np.std(mse_arr_robust)

In [34]:
print('The mean MSE after training the model 50 times after the data is scaled using robust scaler is: ',mean_mse_robust,
      ' and the standard deviation is: ', std_mse_robust)

The mean MSE after training the model 50 times after the data is scaled using robust scaler is:  53.39723753448831  and the standard deviation is:  77.20912119999225


This performance than part A where mean MSE and standard deviation was 58.269382589504986.
However, it haas more standard deviation than part A. The standard deviation in part A was 18.803301390418138 compared to the cuurent standard deviation of 77.20912119999225